<!doctype html><html lang=en><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><title>Integrated Clustering (n = 3)
| Larval Testis X Inactivation</title><link rel=stylesheet href=/larval_gonad_docs/book.min.e0adbac1e0fc6fdc3ef34548c686f58a7e1ba9fb1adf5c9591aa4b7df3653f22.css><link rel=icon href=/larval_gonad_docs/favicon.png type=image/x-icon></head><body><input type=checkbox style=display:none id=menu-control><main class="flex container"><aside class="book-menu fixed"><nav><h2 class=book-brand><a href=https://jfear.github.io/larval_gonad_docs>Larval Testis X Inactivation</a></h2><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/>Data Prep and Quality Control</a><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cell_selection/>Cell Selection</a><ul></ul></li><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cluster_selection/>Selecting Clustering Algorithm</a><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cluster_selection/seurat2/testis1/>Seurat2 Testis 1 Clustering</a></li></ul></li></ul></li><li><a href=/larval_gonad_docs/docs/clustering/>Clustering</a><ul><li><a href=/larval_gonad_docs/docs/clustering/combined_n3/ class=active>Integrated Clustering (n = 3)</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis1/>Testis 1 Individual Clustering</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis2/>Testis 2 Individual Clustering</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis3/>Testis 3 Individual Clustering</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis4/>Testis 4 Individual Clustering</a></li></ul></li></ul></nav><script>(function(){var menu=document.querySelector('aside.book-menu nav')
addEventListener('beforeunload',function(event){localStorage.setItem('menu.scrollTop',menu.scrollTop)});menu.scrollTop=localStorage.getItem('menu.scrollTop')})()</script></aside><div class=book-page><header class="align-center justify-between book-header"><label for=menu-control><img src=/larval_gonad_docs/svg/menu.svg alt=Menu></label>
<strong>Integrated Clustering (n = 3)</strong></header><article class=markdown><h2 id=integrated-clustering-n-3-seurat-v3>Integrated Clustering (n = 3; Seurat v3)</h2><p>After testing both Seurat and Monocle (v2 and v3), I have decided to move forward with Seurat v3.
All method give similar clusters, but Seurat v3 provides more advanced data integration and differential expression testing.
Here I describe the integration of replicates 1, 2, and 3.
This is followed by clustering of cells and annotation of cell type based on expression of known marker genes.</p><h3 id=individual-replicate-clustering>Individual Replicate Clustering</h3><p>I started by clustering each replicate separately.
This gives me an idea of overall data structure and a feeling for how many clusters to expect.
Each replicate had very similar clustering patterns, with very distinct germline and somatic clusters.
See individual clustering documentation for more details.</p><figure><img src=../combined_n3_figures/dimplots-1.png alt="Figure 1. Individual clusters from three replicates (Testis 1, 2, and 3 respectively)." width=100%><figcaption><p><strong>Figure 1. Individual clusters from three replicates (Testis 1, 2, and 3 respectively).</strong></p></figcaption></figure><h3 id=integration-and-clustering>Integration and Clustering</h3><p>Replicates are not common in scRNA-Seq.
We could cluster each replicate separately and then map clusters between replicates.
Or we can combine cells from all replicates and cluster once.
The later would incorporate more data into the clustering callings process, which may improve overall clustering results.</p><p>Here we combine replicates and then cluster.
First, we remove replicate batch effects.
To do this, Seurat provides an alignment method based on canonical correlation analysis (CCA) which tries to improve correlation among samples.
They have made some slight improvements over Seurat v2.
I performed CCA in 30 dimensional space and selected the first 9 dimensions to run KNN clustering.
For community detection (Louvain algorithm) I used a resolution of 0.3 which gives 10 clusters.</p><p>The 2D projections using uniform manifold approximation and projection (UMAP) shows a similar structure as individual replicates (<strong>Figure 1, 2, 3</strong>).
Each replicate is represented in each cluster (<strong>Figure 2A, 3</strong>) and clusters range from 853 (9) to 3,217 (0) cells (<strong>Table 1</strong>).
The contribution of cells to each cluster is based on the proportion of cells from each replicate.
With clusters (0, 1, 2) having proportionally more cells from replicate 3 than other clusters.</p><figure><img src=../combined_n3_figures/umap_reps-1.png alt="Figure 2. UMAP projection of integrated clusters. Each cell (n = 18,792) is plotted in UMAP space. Colors represent either different replciates (panel A) or different cell types after cluster calling using community detection (panel B). Cluster numbering is random." width=100%><figcaption><p><strong>Figure 2. UMAP projection of integrated clusters.</strong> Each cell (n = 18,792) is plotted in UMAP space. Colors represent either different replciates (panel A) or different cell types after cluster calling using community detection (panel B). Cluster numbering is random.</p></figcaption></figure><figure><img src=../combined_n3_figures/umap_reps_split-1.png alt="Figure 3. UMAP projection of integrated clusters split by replicate. Cells from each replicate are plotted separately in UMAP space. Colors represent different cell types after clustering." width=100%><figcaption><p><strong>Figure 3. UMAP projection of integrated clusters split by replicate.</strong> Cells from each replicate are plotted separately in UMAP space. Colors represent different cell types after clustering.</p></figcaption></figure><p><strong>Table 1. Number of cells per cluster.</strong></p><table><thead><tr><th>Cluster</th><th align=right>Num. Cells Per Cluster</th></tr></thead><tbody><tr><td>0</td><td align=right>3,217</td></tr><tr><td>1</td><td align=right>3,149</td></tr><tr><td>2</td><td align=right>2,652</td></tr><tr><td>3</td><td align=right>2,388</td></tr><tr><td>4</td><td align=right>1,510</td></tr><tr><td>5</td><td align=right>1,466</td></tr><tr><td>6</td><td align=right>1,401</td></tr><tr><td>7</td><td align=right>1,376</td></tr><tr><td>8</td><td align=right>870</td></tr><tr><td>9</td><td align=right>853</td></tr></tbody></table><figure><img src=../combined_n3_figures/barplot-cell-per-cluster-number.svg alt="Figure 4. Number of Cells per Replicate per Cluster. The number of cells from each replicate are plotted by cluster." width=100%><figcaption><p><strong>Figure 4. Number of Cells per Replicate per Cluster.</strong> The number of cells from each replicate are plotted by cluster.</p></figcaption></figure><h3 id=cluster-annotation>Cluster Annotation</h3><p>To annotate clusters I used a set of 73 marker genes from the literature (<strong>Figure 5</strong>).
Based on patterns of gene expression, we identified 5 germ cell clusters and 5 somatic cell clusters.</p><ul><li>Spermatogonia (SP: cluster 9)</li><li>Early Primary Spermatocytes (EPS: cluster 5)</li><li>Primary Spermatocytes (PS1: cluster 6)</li><li>Primary Spermatocytes (PS2: cluster 2)</li><li>Primary Spermatocytes (PS3: cluster 0)</li><li>Early Cyst Cells (ECY: cluster 4)</li><li>Cyst Cells (CY1: cluster 1)</li><li>Cyst Cells (CY2: cluster 3)</li><li>Terminal Epithelium (TE: cluster 7)</li><li>Pigment Cells (PC: cluster 8)</li></ul><p>We are able to separate the germ cell lineage into spermatogonia (SP), early primary spermatocytes (EPS), and later primary spermatocytes (PS1, PS2, PS3).
The ordering of the later primary spermatocytes is not clear.
Similarly, in the somatic cyst cell lineage we identified an early cyst cell (ECY) based on <em>tj</em> expression and two later cyst cell types (CY1 and CY2) with CY2 showing the highest expression of <em>eya</em>.</p><figure><img src=../combined_n3_figures/heatmap-all-lit-genes-w-rep.svg alt="Figure 5. Expression patterns of marker genes from the literature by putative cell type. We calculated z-scores for each gene by replicate by cluster after TPM normalization. We plot 73 marker genes from the literature as rows and cluster * replicate as columns. Each marker gene is known to describe a specific cell type (labeled on the right). We used these expression patterns to annotate and order clusters. White dashed lines separate clusters (columns) and gene groups (rows)." width=100%><figcaption><p><strong>Figure 5. Expression patterns of marker genes from the literature by putative cell type.</strong> We calculated z-scores for each gene by replicate by cluster after TPM normalization. We plot 73 marker genes from the literature as rows and cluster * replicate as columns. Each marker gene is known to describe a specific cell type (labeled on the right). We used these expression patterns to annotate and order clusters. White dashed lines separate clusters (columns) and gene groups (rows).</p></figcaption></figure><figure><img src=../combined_n3_figures/umap.svg alt="Figure 6. UMAP projection after annotation. Each cluster was annotated with a putative cell type based on marker gene expression." width=100%><figcaption><p><strong>Figure 6. UMAP projection after annotation.</strong> Each cluster was annotated with a putative cell type based on marker gene expression.</p></figcaption></figure><p>To give a sense of the specificity and localization I plot the various marker genes below.
On the first tab is a Zoom view of the genes I focused on when making annotation decisions.
On the second tab (UMAP) I plot all marker genes for a given cell type.
On the third tab (Violin) I plot the cell distribution for each gene in each cluster.</p><h4 id=spermatogonia-cluster-9>Spermatogonia (Cluster 9)</h4><div class=book-tabs><input type=radio name=tabs-spermatogoina style=display:none id=tabs-spermatogoina-0 checked>
<label for=tabs-spermatogoina-0>UMAP (Zoom)</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/vas_bam-1.png width=100%></figure></div><input type=radio name=tabs-spermatogoina style=display:none id=tabs-spermatogoina-1>
<label for=tabs-spermatogoina-1>UMAP</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_gonia-2.png width=100%></figure></div><input type=radio name=tabs-spermatogoina style=display:none id=tabs-spermatogoina-2>
<label for=tabs-spermatogoina-2>Violin</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_gonia-1.png width=100%></figure></div></div><p></p><h4 id=spermatocytes-clusters-0-2-5-6>Spermatocytes (Clusters 0, 2, 5, 6)</h4><div class=book-tabs><input type=radio name=tabs-spermatocytes style=display:none id=tabs-spermatocytes-0 checked>
<label for=tabs-spermatocytes-0>UMAP (Zoom)</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/aly_nht_can_soti-1.png width=100%></figure></div><input type=radio name=tabs-spermatocytes style=display:none id=tabs-spermatocytes-1>
<label for=tabs-spermatocytes-1>UMAP</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_spermatocytes-2.png width=100%></figure></div><input type=radio name=tabs-spermatocytes style=display:none id=tabs-spermatocytes-2>
<label for=tabs-spermatocytes-2>Violin</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_spermatocytes-1.png width=100%></figure></div></div><p></p><h4 id=cysc-clusters-1-3-4>CySC (Clusters 1, 3, 4)</h4><div class=book-tabs><input type=radio name=tabs-cysc style=display:none id=tabs-cysc-0 checked>
<label for=tabs-cysc-0>UMAP (Zoom)</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/tj_eya-1.png width=100%></figure></div><input type=radio name=tabs-cysc style=display:none id=tabs-cysc-1>
<label for=tabs-cysc-1>UMAP</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_cysc-2.png width=100%></figure></div><input type=radio name=tabs-cysc style=display:none id=tabs-cysc-2>
<label for=tabs-cysc-2>Violin</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_cysc-1.png width=100%></figure></div></div><p></p><h4 id=te-cluster-7>TE (Cluster 7)</h4><div class=book-tabs><input type=radio name=tabs-te style=display:none id=tabs-te-0 checked>
<label for=tabs-te-0>UMAP</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_te-2.png width=100%></figure></div><input type=radio name=tabs-te style=display:none id=tabs-te-1>
<label for=tabs-te-1>Violin</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_te-1.png width=100%></figure></div></div><p></p><h4 id=pigmet-cells-cluster-8>Pigmet Cells (Cluster 8)</h4><div class=book-tabs><input type=radio name=tabs-pc style=display:none id=tabs-pc-0 checked>
<label for=tabs-pc-0>UMAP</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_pc-2.png width=100%></figure></div><input type=radio name=tabs-pc style=display:none id=tabs-pc-1>
<label for=tabs-pc-1>Violin</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_pc-1.png width=100%></figure></div></div><p></p><h4 id=hub>Hub (??)</h4><div class=book-tabs><input type=radio name=tabs-hub style=display:none id=tabs-hub-0 checked>
<label for=tabs-hub-0>UMAP</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_hub-2.png width=100%></figure></div><input type=radio name=tabs-hub style=display:none id=tabs-hub-1>
<label for=tabs-hub-1>Violin</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_hub-1.png width=100%></figure></div></div><p></p><h3 id=differential-expression>Differential Expression</h3><p>Next I look for differentially expressed genes among clusters.
I use a One vs Rest approach to get a list of genes (biomarkers) that are up regulated in each cluster.
Some biomarkers are up regulated in a single cluster (<strong>Figure 7</strong>) while others were up regulated in multiple cluster (<strong>Figure 8</strong>).
Interestingly, some clusters of the germline and somatic cyst cell lineage have very few uniquely up regulated genes (PS2, PS3, and CY1), while the remaining clusters have more than 100.
Perhaps these cells types are intermediates in the developmental program.
Marker genes expressed in multiple cell types tend to be found within the same lineage, with only 367 genes up regulated in the germline and soma.</p><figure><img src=../combined_n3_figures/heatmap-unique-biomarkers-w-rep.svg alt="Figure 7. Biomarkers uniquely up regulated in a single cluster. Using differential expression we identified a set of genes that are significantly up regulated in a specific cluster. For each cluster (columns) we plot z-scores of genes that were uniquely up regulated in that cluster (label on the right and the number of genes). Genes are ordered within each gene group by hierarchical clustering. White dashed lines separate clusters (columns) and gene groups (rows)." width=100%><figcaption><p><strong>Figure 7. Biomarkers uniquely up regulated in a single cluster.</strong> Using differential expression we identified a set of genes that are significantly up regulated in a specific cluster. For each cluster (columns) we plot z-scores of genes that were uniquely up regulated in that cluster (label on the right and the number of genes). Genes are ordered within each gene group by hierarchical clustering. White dashed lines separate clusters (columns) and gene groups (rows).</p></figcaption></figure><figure><img src=../combined_n3_figures/heatmap-multi-biomarkers-w-rep.svg alt="Figure 8. Biomarkers up regulated in multiple clusters. Using differential expression we identified a set of genes that are significantly up regulated in multiple clusters. We group genes that were up regulated in the germ line lineage only (Germ only), in the somatic lineage only (Soma only) and in both germ and somatic cells (Germ and Soma). Within gene group we used hierarchical clustering to sort genes. White dashed lines separate clusters (columns) and gene groups (rows)." width=100%><figcaption><p><strong>Figure 8. Biomarkers up regulated in multiple clusters.</strong> Using differential expression we identified a set of genes that are significantly up regulated in multiple clusters. We group genes that were up regulated in the germ line lineage only (Germ only), in the somatic lineage only (Soma only) and in both germ and somatic cells (Germ and Soma). Within gene group we used hierarchical clustering to sort genes. White dashed lines separate clusters (columns) and gene groups (rows).</p></figcaption></figure><h4 id=biomarkers-on-the-literature-gene-list>Biomarkers on the Literature Gene List</h4><p>Using the biomakers list I looked to see what fraction of genes overlap our literature gene list.
Every putative cell type had at least a few genes from the literature gene list.
CY1 and CY2 had the lowest representation of literature genes on the biomaker list (7% and 19% respectively).
TE and PC had the best representation (100%), but also had the smallest number of genes.
As expected, the EPS cluster had the early tTAF (<em>aly</em>, <em>can</em>), while the later staged germ cells PS1, PS2, and PS3 had the later stage tTAFs (<em>dj</em>, <em>ocn</em>, <em>fzo</em>).</p><div class=book-tabs><input type=radio name=tabs-Overlap style=display:none id=tabs-Overlap-0 checked>
<label for=tabs-Overlap-0>SP</label><div class="book-tabs-content markdown-inner">43% Overlap</p><table border=1 class=dataframe><thead><tr style=text-align:right><th></th><th></th><th colspan=2>Literature Gene</th></tr><tr style=text-align:right><th></th><th></th><th>False</th><th>True</th></tr></thead><tbody><tr><th rowspan=2>Biomarker</th><th>False</th><td>16,528</td><td>9</td></tr><tr><th>True</th><td>1,227</td><td>7</td></tr></tbody></table><ul><li>Lit Gene In Biomarker List: bam, Rbp4, Phf7, p53, AGO3, Prosalpha6, vas</li><li>Not In Biomarker List: esg, neur, nos, bgcn, spi, Dad, fest, dpr17, tut</li></ul></div><input type=radio name=tabs-Overlap style=display:none id=tabs-Overlap-1>
<label for=tabs-Overlap-1>EPS</label><div class="book-tabs-content markdown-inner">43% Overlap</p><table border=1 class=dataframe><thead><tr style=text-align:right><th></th><th></th><th colspan=2>Literature Gene</th></tr><tr style=text-align:right><th></th><th></th><th>False</th><th>True</th></tr></thead><tbody><tr><th rowspan=2>Biomarker</th><th>False</th><td>16,626</td><td>13</td></tr><tr><th>True</th><td>1,122</td><td>10</td></tr></tbody></table><ul><li>Lit Gene In Biomarker List: aub, sa, aly, can, Taf12L, tomb, kmg, CG3927, tbrd-1, nht</li><li>Not In Biomarker List: CycA, CycB, Mst87F, bol, fzo, mia, dj, Reepl1, oys, topi, d-cup, bb8, ocn</li></ul></div><input type=radio name=tabs-Overlap style=display:none id=tabs-Overlap-2>
<label for=tabs-Overlap-2>PS1</label><div class="book-tabs-content markdown-inner">43% Overlap</p><table border=1 class=dataframe><thead><tr style=text-align:right><th></th><th></th><th colspan=2>Literature Gene</th></tr><tr style=text-align:right><th></th><th></th><th>False</th><th>True</th></tr></thead><tbody><tr><th rowspan=2>Biomarker</th><th>False</th><td>16,058</td><td>13</td></tr><tr><th>True</th><td>1,690</td><td>10</td></tr></tbody></table><ul><li>Lit Gene In Biomarker List: CycB, Mst87F, bol, fzo, dj, Reepl1, CG3927, d-cup, bb8, ocn</li><li>Not In Biomarker List: aub, CycA, sa, aly, can, mia, Taf12L, tomb, kmg, oys, topi, tbrd-1, nht</li></ul></div><input type=radio name=tabs-Overlap style=display:none id=tabs-Overlap-3>
<label for=tabs-Overlap-3>PS2</label><div class="book-tabs-content markdown-inner">47% Overlap</p><table border=1 class=dataframe><thead><tr style=text-align:right><th></th><th></th><th colspan=2>Literature Gene</th></tr><tr style=text-align:right><th></th><th></th><th>False</th><th>True</th></tr></thead><tbody><tr><th rowspan=2>Biomarker</th><th>False</th><td>16,295</td><td>12</td></tr><tr><th>True</th><td>1,453</td><td>11</td></tr></tbody></table><ul><li>Lit Gene In Biomarker List: CycB, Mst87F, bol, fzo, dj, Reepl1, Taf12L, CG3927, d-cup, bb8, ocn</li><li>Not In Biomarker List: aub, CycA, sa, aly, can, mia, tomb, kmg, oys, topi, tbrd-1, nht</li></ul></div><input type=radio name=tabs-Overlap style=display:none id=tabs-Overlap-4>
<label for=tabs-Overlap-4>PS3</label><div class="book-tabs-content markdown-inner">21% Overlap</p><table border=1 class=dataframe><thead><tr style=text-align:right><th></th><th></th><th colspan=2>Literature Gene</th></tr><tr style=text-align:right><th></th><th></th><th>False</th><th>True</th></tr></thead><tbody><tr><th rowspan=2>Biomarker</th><th>False</th><td>17,370</td><td>18</td></tr><tr><th>True</th><td>378</td><td>5</td></tr></tbody></table><ul><li>Lit Gene In Biomarker List: Mst87F, dj, Taf12L, CG3927, ocn</li><li>Not In Biomarker List: aub, CycA, CycB, sa, aly, bol, can, fzo, mia, Reepl1, tomb, kmg, oys, topi, d-cup, bb8, tbrd-1, nht</li></ul></div><input type=radio name=tabs-Overlap style=display:none id=tabs-Overlap-5>
<label for=tabs-Overlap-5>EYC</label><div class="book-tabs-content markdown-inner">36% Overlap</p><table border=1 class=dataframe><thead><tr style=text-align:right><th></th><th></th><th colspan=2>Literature Gene</th></tr><tr style=text-align:right><th></th><th></th><th>False</th><th>True</th></tr></thead><tbody><tr><th rowspan=2>Biomarker</th><th>False</th><td>17,083</td><td>17</td></tr><tr><th>True</th><td>662</td><td>9</td></tr></tbody></table><ul><li>Lit Gene In Biomarker List: tj, bnb, ImpL2, vn, Nrt, piwi, Wnt4, fax, kek1</li><li>Not In Biomarker List: eya, EcR, robo2, sev, so, usp, zfh1, fng, gbb, spict, sano, Cht5, Efa6, Nlg3, rdo, puc, br</li></ul></div><input type=radio name=tabs-Overlap style=display:none id=tabs-Overlap-6>
<label for=tabs-Overlap-6>CY1</label><div class="book-tabs-content markdown-inner">7% Overlap</p><table border=1 class=dataframe><thead><tr style=text-align:right><th></th><th></th><th colspan=2>Literature Gene</th></tr><tr style=text-align:right><th></th><th></th><th>False</th><th>True</th></tr></thead><tbody><tr><th rowspan=2>Biomarker</th><th>False</th><td>17,495</td><td>24</td></tr><tr><th>True</th><td>250</td><td>2</td></tr></tbody></table><ul><li>Lit Gene In Biomarker List: bnb, fax</li><li>Not In Biomarker List: eya, EcR, tj, ImpL2, robo2, sev, so, usp, vn, Nrt, zfh1, piwi, Wnt4, fng, kek1, gbb, spict, sano, Cht5, Efa6, Nlg3, rdo, puc, br</li></ul></div><input type=radio name=tabs-Overlap style=display:none id=tabs-Overlap-7>
<label for=tabs-Overlap-7>CY2</label><div class="book-tabs-content markdown-inner">19% Overlap</p><table border=1 class=dataframe><thead><tr style=text-align:right><th></th><th></th><th colspan=2>Literature Gene</th></tr><tr style=text-align:right><th></th><th></th><th>False</th><th>True</th></tr></thead><tbody><tr><th rowspan=2>Biomarker</th><th>False</th><td>16,929</td><td>21</td></tr><tr><th>True</th><td>816</td><td>5</td></tr></tbody></table><ul><li>Lit Gene In Biomarker List: bnb, fax, kek1, Nlg3, rdo</li><li>Not In Biomarker List: eya, EcR, tj, ImpL2, robo2, sev, so, usp, vn, Nrt, zfh1, piwi, Wnt4, fng, gbb, spict, sano, Cht5, Efa6, puc, br</li></ul></div><input type=radio name=tabs-Overlap style=display:none id=tabs-Overlap-8>
<label for=tabs-Overlap-8>TE</label><div class="book-tabs-content markdown-inner">100% Overlap</p><table border=1 class=dataframe><thead><tr style=text-align:right><th></th><th></th><th colspan=2>Literature Gene</th></tr><tr style=text-align:right><th></th><th></th><th>False</th><th>True</th></tr></thead><tbody><tr><th rowspan=2>Biomarker</th><th>False</th><td>17,147</td><td>0</td></tr><tr><th>True</th><td>618</td><td>6</td></tr></tbody></table><ul><li>Lit Gene In Biomarker List: abd-A, Abd-B, cv-2, N, nord, Piezo</li></ul></div><input type=radio name=tabs-Overlap style=display:none id=tabs-Overlap-9>
<label for=tabs-Overlap-9>PC</label><div class="book-tabs-content markdown-inner">100% Overlap</p><table border=1 class=dataframe><thead><tr style=text-align:right><th></th><th></th><th colspan=2>Literature Gene</th></tr><tr style=text-align:right><th></th><th></th><th>False</th><th>True</th></tr></thead><tbody><tr><th rowspan=2>Biomarker</th><th>False</th><td>17,054</td><td>0</td></tr><tr><th>True</th><td>715</td><td>2</td></tr></tbody></table><ul><li>Lit Gene In Biomarker List: ems, Sox100B</li></ul></div></div><p></p><h4 id=potentially-miss-expressed-literature-genes>Potentially Miss Expressed Literature Genes</h4><p>Next I wanted to look for genes that appear as a different marker on the literature gene list.
This is complicated to look at because many of the markers are in multiple cell types.
I only include clusters that had marker gene mapping to another cell type.
There are only a few instances of a germline marker gene being found on the biomarker list of a somatic cell type.
It is hard to know if this is because of cell contamination or if the gene is expressed in that cell type.</p><p><div class=book-tabs><input type=radio name=tabs-missexpressed style=display:none id=tabs-missexpressed-0 checked>
<label for=tabs-missexpressed-0>SP</label><div class="book-tabs-content markdown-inner"><ul><li>aub (ES, MS)</li><li>Taf12L (ES, MS, LS)</li></ul></div><input type=radio name=tabs-missexpressed style=display:none id=tabs-missexpressed-1>
<label for=tabs-missexpressed-1>EPS</label><div class="book-tabs-content markdown-inner"><ul><li>AGO3 (SP)</li></ul></div><input type=radio name=tabs-missexpressed style=display:none id=tabs-missexpressed-2>
<label for=tabs-missexpressed-2>CY2</label><div class="book-tabs-content markdown-inner"><ul><li>cv-2 (TE)</li><li>Dad (SP)</li><li>Prosalpha6 (ES)</li></ul></div><input type=radio name=tabs-missexpressed style=display:none id=tabs-missexpressed-3>
<label for=tabs-missexpressed-3>TE</label><div class="book-tabs-content markdown-inner"><ul><li>ImpL2 (C, EC)</li><li>robo2 (H, C)</li><li>zfh1 (C, EC)</li><li>spi (SP, ES)</li><li>fax (C, EC, MC)</li><li>kek1 (C, EC, MC, LC)</li><li>Dad (SP)</li><li>gbb (H, C, EC)</li><li>Socs36E (H, C)</li><li>puc (C, EC)</li><li>br (C, EC)</li></ul></div><input type=radio name=tabs-missexpressed style=display:none id=tabs-missexpressed-4>
<label for=tabs-missexpressed-4>PC</label><div class="book-tabs-content markdown-inner"><ul><li>abd-A (H, C, EC)</li><li>EcR (C, EC)</li><li>robo2 (H, C)</li><li>vn (C, EC)</li><li>zfh1 (C, EC)</li><li>N (H, C, TE)</li><li>spi (SP, ES)</li><li>fax (C, EC, MC)</li><li>kek1 (C, EC, MC, LC)</li><li>spict (EC)</li><li>Socs36E (H, C)</li><li>puc (C, EC)</li><li>br (C, EC)</li></ul></div></div><p></p><em>Note abbreviations in parentheses are based on Miriam&rsquo;s literature gene table.</em></p><h4 id=biomarker-gene-ontology>Biomarker Gene Ontology</h4><p>To get a better understanding of the underlying biological processes driving the differences among clusters I looked at gene ontology.
SP and EPS have genes associated with transcription and translation.
EPS was also significant for chromatin organization.
PS1, PS2, and P3 show genes associated with sperm development.
With genes up regulated in PS3 almost entirely associated with sperm production.
The cyst cell lineage (ECY, CY1, and CY2) show evidence of genes associated with translation.
The TE and PC clusters show genes associated with morphogenesis and growth.</p><div class=book-tabs><input type=radio name="tabs-Gene Ontology" style=display:none id="tabs-Gene Ontology-0" checked>
<label for="tabs-Gene Ontology-0">SP</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/sp-1.png width=100%></figure></div><input type=radio name="tabs-Gene Ontology" style=display:none id="tabs-Gene Ontology-1">
<label for="tabs-Gene Ontology-1">EPS</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/eps-1.png width=100%></figure></div><input type=radio name="tabs-Gene Ontology" style=display:none id="tabs-Gene Ontology-2">
<label for="tabs-Gene Ontology-2">PS1</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/ps1-1.png width=100%></figure></div><input type=radio name="tabs-Gene Ontology" style=display:none id="tabs-Gene Ontology-3">
<label for="tabs-Gene Ontology-3">PS2</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/ps2-1.png width=100%></figure></div><input type=radio name="tabs-Gene Ontology" style=display:none id="tabs-Gene Ontology-4">
<label for="tabs-Gene Ontology-4">PS3</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/ps3-1.png width=100%></figure></div><input type=radio name="tabs-Gene Ontology" style=display:none id="tabs-Gene Ontology-5">
<label for="tabs-Gene Ontology-5">ECY</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/ecy-1.png width=100%></figure></div><input type=radio name="tabs-Gene Ontology" style=display:none id="tabs-Gene Ontology-6">
<label for="tabs-Gene Ontology-6">CY1</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/cy1-1.png width=100%></figure></div><input type=radio name="tabs-Gene Ontology" style=display:none id="tabs-Gene Ontology-7">
<label for="tabs-Gene Ontology-7">CY2</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/cy2-1.png width=100%></figure></div><input type=radio name="tabs-Gene Ontology" style=display:none id="tabs-Gene Ontology-8">
<label for="tabs-Gene Ontology-8">TE</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/te-1.png width=100%></figure></div><input type=radio name="tabs-Gene Ontology" style=display:none id="tabs-Gene Ontology-9">
<label for="tabs-Gene Ontology-9">PC</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/pc-1.png width=100%></figure></div></div><p></p></article><div class="book-footer justify-between"><div><a href=https://github.com/jfear/larval_gonad_docs/commit/fe5851a8ffba43b3537fad01bd5f481b3fdcf5ee title="Last modified Jun 28, 2019 by Justin M. Fear" target=_blank rel=noopener><img src=/larval_gonad_docs/svg/calendar.svg alt=Changed> Jun 28, 2019</a></div><div><a href=https://github.com/jfear/larval_gonad_docs/edit/master/content/docs/clustering/combined_n3.md target=_blank rel=noopener><img src=/larval_gonad_docs/svg/edit.svg alt=Edit> Edit this page</a></div></div></div><aside class="book-toc level-4 fixed"><nav id=TableOfContents><ul><li><ul><li><a href=#integrated-clustering-n-3-seurat-v3>Integrated Clustering (n = 3; Seurat v3)</a><ul><li><a href=#individual-replicate-clustering>Individual Replicate Clustering</a></li><li><a href=#integration-and-clustering>Integration and Clustering</a></li><li><a href=#cluster-annotation>Cluster Annotation</a><ul><li><a href=#spermatogonia-cluster-9>Spermatogonia (Cluster 9)</a></li><li><a href=#spermatocytes-clusters-0-2-5-6>Spermatocytes (Clusters 0, 2, 5, 6)</a></li><li><a href=#cysc-clusters-1-3-4>CySC (Clusters 1, 3, 4)</a></li><li><a href=#te-cluster-7>TE (Cluster 7)</a></li><li><a href=#pigmet-cells-cluster-8>Pigmet Cells (Cluster 8)</a></li><li><a href=#hub>Hub (??)</a></li></ul></li><li><a href=#differential-expression>Differential Expression</a><ul><li><a href=#biomarkers-on-the-literature-gene-list>Biomarkers on the Literature Gene List</a></li><li><a href=#potentially-miss-expressed-literature-genes>Potentially Miss Expressed Literature Genes</a></li><li><a href=#biomarker-gene-ontology>Biomarker Gene Ontology</a></li></ul></li></ul></li></ul></li></ul></nav></aside></main></body></html>