<!doctype html><html lang=en><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><title>Testis Integrated Clustering (n = 3)
| Larval Testis X Inactivation</title><link rel=stylesheet href=/larval_gonad_docs/book.min.e0adbac1e0fc6fdc3ef34548c686f58a7e1ba9fb1adf5c9591aa4b7df3653f22.css><link rel=icon href=/larval_gonad_docs/favicon.png type=image/x-icon></head><body><input type=checkbox style=display:none id=menu-control><main class="flex container"><aside class="book-menu fixed"><nav><h2 class=book-brand><a href=https://jfear.github.io/larval_gonad_docs>Larval Testis X Inactivation</a></h2><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/>Data Prep and Quality Control</a><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cell_selection/>Cell Selection</a><ul></ul></li><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cluster_selection/>Cell Clustering</a><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cluster_selection/seurat2/testis1/>Seurat2 Testis 1 Clustering</a></li></ul></li></ul></li><li><a href=/larval_gonad_docs/docs/clustering/>Clustering</a><ul><li><a href=/larval_gonad_docs/docs/clustering/combined_n3/ class=active>Testis Integrated Clustering (n = 3)</a></li></ul></li></ul></nav><script>(function(){var menu=document.querySelector('aside.book-menu nav')
addEventListener('beforeunload',function(event){localStorage.setItem('menu.scrollTop',menu.scrollTop)});menu.scrollTop=localStorage.getItem('menu.scrollTop')})()</script></aside><div class=book-page><header class="align-center justify-between book-header"><label for=menu-control><img src=/larval_gonad_docs/svg/menu.svg alt=Menu></label>
<strong>Testis Integrated Clustering (n = 3)</strong></header><article class=markdown><h2 id=testis-integrated-clustering-n-3-seurat-v3>Testis Integrated Clustering (n = 3; Seurat v3)</h2><h3 id=umap-projection-clustering-from-individual-replicates>UMAP Projection Clustering From Individual Replicates</h3><p>First I clustered each replicate separately using a replicate specific workflow.
For consistency, I called 12-13 clusters for each replicate.
Cluster numbers and colors do not correspond.
For more details see individual clustering analysis.</p><figure><img src=../combined_n3_figures/dimplots-1.png width=100%></figure><h3 id=umap-projection-clustering-from-integrated-replicates>UMAP Projection Clustering From Integrated Replicates</h3><p>Integrated analysis begins by taking normalized reads from each replicate and aligning cells using canonical correlation analysis (CCA).
Cells were clustered using Seurat v3.
Individual sample clustering shows better separation of germ cells and somatic cells, but overall structure is maintained in the integrated analysis.
The largest cluster (0) had 3,217 cells while the smallest cluster (9) had 853 cells.
Cells from each replicate are present in each cluster.</p><figure><img src=../combined_n3_figures/umap_reps-1.png width=100%></figure><figure><img src=../combined_n3_figures/umap_reps_split-1.png width=100%></figure><p><strong>Table 1. Number of cells per integrated cluster.</strong></p><table><thead><tr><th>Cluster</th><th align=right>Num. Cells Per Cluster</th></tr></thead><tbody><tr><td>0</td><td align=right>3,217</td></tr><tr><td>1</td><td align=right>3,149</td></tr><tr><td>2</td><td align=right>2,652</td></tr><tr><td>3</td><td align=right>2,388</td></tr><tr><td>4</td><td align=right>1,510</td></tr><tr><td>5</td><td align=right>1,466</td></tr><tr><td>6</td><td align=right>1,401</td></tr><tr><td>7</td><td align=right>1,376</td></tr><tr><td>8</td><td align=right>870</td></tr><tr><td>9</td><td align=right>853</td></tr></tbody></table><h3 id=literature-genes>Literature Genes</h3><p>To annotate clusters I am using genes from the literature.
Plotted below are UMAP projects for genes from the literature.
I also have violin plots showing the distribution of expression for each gene.
For some cell types I show a &ldquo;Zoomed&rdquo; view of a few key genes.</p><h4 id=spermatogonia-cluster-9>Spermatogonia (Cluster 9)</h4><div class=book-tabs><input type=radio name=tabs-spermatogoina style=display:none id=tabs-spermatogoina-0 checked>
<label for=tabs-spermatogoina-0>UMAP (Zoom)</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/vas_bam-1.png width=100%></figure></div><input type=radio name=tabs-spermatogoina style=display:none id=tabs-spermatogoina-1>
<label for=tabs-spermatogoina-1>UMAP</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_gonia-2.png width=100%></figure></div><input type=radio name=tabs-spermatogoina style=display:none id=tabs-spermatogoina-2>
<label for=tabs-spermatogoina-2>Violin</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_gonia-1.png width=100%></figure></div></div><p></p><h4 id=spermatocytes-clusters-0-2-5-6>Spermatocytes (Clusters 0, 2, 5, 6)</h4><div class=book-tabs><input type=radio name=tabs-spermatocytes style=display:none id=tabs-spermatocytes-0 checked>
<label for=tabs-spermatocytes-0>UMAP (Zoom)</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/aly_nht_can_soti-1.png width=100%></figure></div><input type=radio name=tabs-spermatocytes style=display:none id=tabs-spermatocytes-1>
<label for=tabs-spermatocytes-1>UMAP</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_spermatocytes-2.png width=100%></figure></div><input type=radio name=tabs-spermatocytes style=display:none id=tabs-spermatocytes-2>
<label for=tabs-spermatocytes-2>Violin</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_spermatocytes-1.png width=100%></figure></div></div><p></p><h4 id=cysc-clusters-1-3-4>CySC (Clusters 1, 3, 4)</h4><div class=book-tabs><input type=radio name=tabs-cysc style=display:none id=tabs-cysc-0 checked>
<label for=tabs-cysc-0>UMAP (Zoom)</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/tj_eya-1.png width=100%></figure></div><input type=radio name=tabs-cysc style=display:none id=tabs-cysc-1>
<label for=tabs-cysc-1>UMAP</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_cysc-2.png width=100%></figure></div><input type=radio name=tabs-cysc style=display:none id=tabs-cysc-2>
<label for=tabs-cysc-2>Violin</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_cysc-1.png width=100%></figure></div></div><p></p><h4 id=te-cluster-7>TE (Cluster 7)</h4><div class=book-tabs><input type=radio name=tabs-te style=display:none id=tabs-te-0 checked>
<label for=tabs-te-0>UMAP</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_te-2.png width=100%></figure></div><input type=radio name=tabs-te style=display:none id=tabs-te-1>
<label for=tabs-te-1>Violin</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_te-1.png width=100%></figure></div></div><p></p><h4 id=pigmet-cells-cluster-8>Pigmet Cells (Cluster 8)</h4><div class=book-tabs><input type=radio name=tabs-pc style=display:none id=tabs-pc-0 checked>
<label for=tabs-pc-0>UMAP</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_pc-2.png width=100%></figure></div><input type=radio name=tabs-pc style=display:none id=tabs-pc-1>
<label for=tabs-pc-1>Violin</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_pc-1.png width=100%></figure></div></div><p></p><h4 id=hub>Hub (??)</h4><div class=book-tabs><input type=radio name=tabs-hub style=display:none id=tabs-hub-0 checked>
<label for=tabs-hub-0>UMAP</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_hub-2.png width=100%></figure></div><input type=radio name=tabs-hub style=display:none id=tabs-hub-1>
<label for=tabs-hub-1>Violin</label><div class="book-tabs-content markdown-inner"><figure><img src=../combined_n3_figures/heatmap_hub-1.png width=100%></figure></div></div><p></p></article><div class="book-footer justify-between"><div><a href=https://github.com/jfear/larval_gonad_docs/commit/e0647ecac483835f568a64b87a80b695d21bcf56 title="Last modified Jun 24, 2019 by Justin M. Fear" target=_blank rel=noopener><img src=/larval_gonad_docs/svg/calendar.svg alt=Changed> Jun 24, 2019</a></div><div><a href=https://github.com/jfear/larval_gonad_docs/edit/master/content/docs/clustering/combined_n3.md target=_blank rel=noopener><img src=/larval_gonad_docs/svg/edit.svg alt=Edit> Edit this page</a></div></div></div><aside class="book-toc level-4 fixed"><nav id=TableOfContents><ul><li><ul><li><a href=#testis-integrated-clustering-n-3-seurat-v3>Testis Integrated Clustering (n = 3; Seurat v3)</a><ul><li><a href=#umap-projection-clustering-from-individual-replicates>UMAP Projection Clustering From Individual Replicates</a></li><li><a href=#umap-projection-clustering-from-integrated-replicates>UMAP Projection Clustering From Integrated Replicates</a></li><li><a href=#literature-genes>Literature Genes</a><ul><li><a href=#spermatogonia-cluster-9>Spermatogonia (Cluster 9)</a></li><li><a href=#spermatocytes-clusters-0-2-5-6>Spermatocytes (Clusters 0, 2, 5, 6)</a></li><li><a href=#cysc-clusters-1-3-4>CySC (Clusters 1, 3, 4)</a></li><li><a href=#te-cluster-7>TE (Cluster 7)</a></li><li><a href=#pigmet-cells-cluster-8>Pigmet Cells (Cluster 8)</a></li><li><a href=#hub>Hub (??)</a></li></ul></li></ul></li></ul></li></ul></nav></aside></main></body></html>