<!doctype html><html lang=en><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><title>Intronless Gene Expression
| Larval Testis X Inactivation</title><link rel=stylesheet href=/larval_gonad_docs/book.min.e0adbac1e0fc6fdc3ef34548c686f58a7e1ba9fb1adf5c9591aa4b7df3653f22.css><link rel=icon href=/larval_gonad_docs/favicon.png type=image/x-icon><link rel=alternate type=application/rss+xml href=https://jfear.github.io/larval_gonad_docs/docs/intronless_genes/index.xml title="Larval Testis X Inactivation"></head><body><input type=checkbox style=display:none id=menu-control><main class="flex container"><aside class="book-menu fixed"><nav><h2 class=book-brand><a href=https://jfear.github.io/larval_gonad_docs>Larval Testis X Inactivation</a></h2><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/>Data Prep and Quality Control</a><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cell_selection/>Cell Selection</a><ul></ul></li><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cluster_selection/>Selecting Clustering Algorithm</a><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cluster_selection/seurat2/testis1/>Seurat2 Testis 1 Clustering</a></li></ul></li></ul></li><li><a href=/larval_gonad_docs/docs/clustering/>Clustering</a><ul><li><a href=/larval_gonad_docs/docs/clustering/combined_n3/>Integrated Clustering (n = 3)</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis1/>Testis 1 Individual Clustering</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis2/>Testis 2 Individual Clustering</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis3/>Testis 3 Individual Clustering</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis4/>Testis 4 Individual Clustering</a></li></ul></li><li><a href=/larval_gonad_docs/docs/intronless_genes/ class=active>Intronless Gene Expression</a><ul></ul></li><li><a href=/larval_gonad_docs/docs/gene_movement/>Neo X Gene Movement</a><ul><li><a href=/larval_gonad_docs/docs/gene_movement/dpse_gonia_vs_cytes/>D. Pseudoobscura (SP vs Cytes)</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dpse_gonia_vs_eps/>D. Pseudoobscura (SP vs EPS)</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dpse_gonia_vs_ps/>D. Pseudoobscura (SP vs PS)</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dpse_eps_vs_ps/>D. Pseudoobscura (EPS vs PS)</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dwil_gonia_vs_cytes/>D. willistoni (SP vs Cytes)</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dwil_gonia_vs_eps/>D. willistoni (SP vs EPS)</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dwil_gonia_vs_ps/>D. willistoni (SP vs PS)</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dwil_eps_vs_ps/>D. willistoni (EPS vs PS)</a></li></ul></li></ul></nav><script>(function(){var menu=document.querySelector('aside.book-menu nav')
addEventListener('beforeunload',function(event){localStorage.setItem('menu.scrollTop',menu.scrollTop)});menu.scrollTop=localStorage.getItem('menu.scrollTop')})()</script></aside><div class=book-page><header class="align-center justify-between book-header"><label for=menu-control><img src=/larval_gonad_docs/svg/menu.svg alt=Menu></label>
<strong>Intronless Gene Expression</strong></header><article class=markdown><p><span style="font-size:3em;font-weight:'strong';color:red">WARNING: Preliminary Analysis</span></p><h2 id=intronless-gene-expression>Intronless Gene Expression</h2><p>Here I look at intronless genes in primary spermatocytes.
The use of intronless genes are important in two areas:
(1) new genes from retro-transposition are likely to have no introns,
(2) genes without introns are easier to transcribe.</p><h3 id=how-are-intronless-genes-expressed-in-primary-spermatocytes>How are intronless genes expressed in primary spermatocytes?</h3><figure><img src=./x_escapers_and_intronless_genes.svg alt="Figure 1. Percent of spermatocyte cells with expression of intronless genes by chromosome arm." width=100%><figcaption><p><b>Figure 1. Percent of spermatocyte cells with expression of intronless genes by chromosome arm.</b></p></figcaption></figure><p>To get a general view of intronless expression, I looked at the percent
of primary spermatocyte cells (EPS, PS1, PS2, PS3) that express intron
containing or intronless genes (<strong>Figure 1</strong>).
Spermatocyte biased genes (top row) are expressed in a higher percentage of spermatocyte cells.
While intronless genes are expressed in fewer cells than intron containing genes irrespective of chromosome arm.
However, the number of intronless genes are up regulated in spermatocytes is higher than expected (<strong>Table 1</strong>).
The overall enrichment of intronless genes in spermatocytes can be see by looking by looking at expression patterns (<strong>Figure 2</strong>).</p><p><strong>Table 1. Spermatocyte biased expressed genes are enriched for intronless genes.</strong></p><table><thead><tr><th>intronless</th><th>NS</th><th>cyte</th><th>gonia</th></tr></thead><tbody><tr><td>False</td><td>64<sup>*</sup></td><td>219<sup>*</sup></td><td>586<sup>‚Ä†</sup></td></tr><tr><td>True</td><td>44<sup>‚Ä†</sup></td><td>141<sup>‚Ä†</sup></td><td>65<sup>*</sup></td></tr></tbody></table><p><span>ùõò^2: 137.1037, p-value: 0.0000, df: 2</span><br><span><sup>*</sup>Depleted: Fewer genes than expected.</span><br><span><sup>‚Ä†</sup>Enriched: More genes than expected.</span></p><figure><img src=./x_escapers_and_intronless_genes_heatmap.svg alt="Figure 2. Intronless gene expression patterns. Z-scores of TPM normalized expression for 2,768 intronless genes." width=100%><figcaption><p><b>Figure 2. Intronless gene expression patterns.</b> Z-scores of TPM normalized expression for 2,768 intronless genes.</p></figcaption></figure><h3 id=are-intronless-genes-enriched-in-x-chromosome-escapers>Are intronless genes enriched in X chromosome escapers?</h3><p>Next were are interested to see the relationships among X chromosome, spermatocyte biased gene expression, and intronless genes.
This question requires statistical modeling because the X chromosome is already depleted of intronless genes (<strong>Table 2</strong>).
We used logistic regression to ask if spermatocyte bias and X chromosome status could predict if a gene was intronless.
First, we looked at the main effects model (<strong>Figure 3</strong>).
Similar to above results, we see that genes with spermatocyte bias expression have an increased likelihood of being intronless.
However, we do not see an effect of X chromosome localization independent of spermatocyte expression.
If we then add the interaction term (<strong>Figure 4</strong>), we see a significant decrease in likelihood of intronless genes being spermatocyte biased and on the X.</p><p><strong>Table 2. The X chromosome is depleted for intronless.</strong></p><table><thead><tr><th>intronless</th><th>2L</th><th>2R</th><th>3L</th><th>3R</th><th>4</th><th>X</th><th>Y</th></tr></thead><tbody><tr><td>False</td><td>2,406<sup>*</sup></td><td>2,604</td><td>2,490</td><td>3,183<sup>‚Ä†</sup></td><td>96<sup>‚Ä†</sup></td><td>2,036<sup>‚Ä†</sup></td><td>34<sup>*</sup></td></tr><tr><td>True</td><td>1,095<sup>‚Ä†</sup></td><td>1,024</td><td>974</td><td>1,018<sup>*</sup></td><td>15<sup>*</sup></td><td>640<sup>*</sup></td><td>79<sup>‚Ä†</sup></td></tr></tbody></table><p><span>ùõò^2: 179.5973, p-value: 0.0000, df: 6</span><br><span><sup>*</sup>Depleted: Fewer genes than expected.</span><br><span><sup>‚Ä†</sup>Enriched: More genes than expected.</span></p><figure><img src=./x_escapers_and_intronless_genes_main_effects.png alt="Figure 3. Logistic regression results of the main effects model. Logit(intronless = cyte_biased &#43; X chromosome)" width=100%><figcaption><p><b>Figure 3. Logistic regression results of the main effects model.</b> Logit(intronless = cyte_biased + X chromosome)</p></figcaption></figure><figure><img src=./x_escapers_and_intronless_genes_full.png alt="Figure 4. Logistic regression of the full model. Logit(intronless = cyte_biased &#43; X chromosome &#43; cyte_biased * X chromosome)" width=100%><figcaption><p><b>Figure 4. Logistic regression of the full model.</b> Logit(intronless = cyte_biased + X chromosome + cyte_biased * X chromosome)</p></figcaption></figure></article><div class="book-footer justify-between"><div><a href=https://github.com/jfear/larval_gonad_docs/commit/67a2b08084434cf6370fee031529f71976455bbf title="Last modified Jul 24, 2019 by Justin Fear" target=_blank rel=noopener><img src=/larval_gonad_docs/svg/calendar.svg alt=Changed> Jul 24, 2019</a></div><div><a href=https://github.com/jfear/larval_gonad_docs/edit/master/content/docs/intronless_genes/_index.md target=_blank rel=noopener><img src=/larval_gonad_docs/svg/edit.svg alt=Edit> Edit this page</a></div></div></div><aside class="book-toc level-4 fixed"><nav id=TableOfContents><ul><li><ul><li><a href=#intronless-gene-expression>Intronless Gene Expression</a><ul><li><a href=#how-are-intronless-genes-expressed-in-primary-spermatocytes>How are intronless genes expressed in primary spermatocytes?</a></li><li><a href=#are-intronless-genes-enriched-in-x-chromosome-escapers>Are intronless genes enriched in X chromosome escapers?</a></li></ul></li></ul></li></ul></nav></aside></main></body></html>