<!doctype html><html lang=en><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><title>10X Cell Selection
| Larval Testis X Inactivation</title><link rel=stylesheet href=/book.min.147a61fa43a9a449c9e241607bc0f659542afcc7be8431be47d114da65fceb40.css><link rel=icon href=/favicon.png type=image/x-icon><link rel=alternate type=application/rss+xml href=https://example.com/docs/scrnaseq_prep/cell_selection/index.xml title="Larval Testis X Inactivation"></head><body><input type=checkbox style=display:none id=menu-control><main class="flex container"><aside class="book-menu fixed"><nav><h2 class=book-brand><a href=https://example.com/>Larval Testis X Inactivation</a></h2><ul><li><a href=/docs/scrnaseq_prep/>Data Prep and Quality Control</a><ul><li><a href=/docs/scrnaseq_prep/cell_selection/ class=active>10X Cell Selection</a><ul></ul></li><li><a href=/docs/scrnaseq_prep/clustering/>Cell Clustering</a><ul></ul></li></ul></li></ul></nav><script>(function(){var menu=document.querySelector('aside.book-menu nav')
addEventListener('beforeunload',function(event){localStorage.setItem('menu.scrollTop',menu.scrollTop)});menu.scrollTop=localStorage.getItem('menu.scrollTop')})()</script></aside><div class=book-page><header class="align-center justify-between book-header"><label for=menu-control><img src=/svg/menu.svg alt=Menu></label>
<strong>10X Cell Selection</strong></header><article class=markdown><figure><img src="https://cdn.technologynetworks.com/tn/images/thumbs/webp/640_360/10x-genomics-extends-their-application-portfolio-305346.webp?v=9720678"><figcaption><h4>The 10X Library Preparation consists of mixing individual cells with individual gel beads.</h4></figcaption></figure><h2 id=barcode-rank-plots>Barcode Rank Plots</h2><p>After sequencing, the first step in scRNA-Seq analysis is identifying cells.
During library preparation, individual cells are mixed with individual beads.
Each bead contains a unique barcode which is used for demultiplexing.
The goal of cell selection is to find beads that had 1 and only 1 cell.
Because cells are loaded in a double Poisson rate, the majority of beads will be empty.
However, these empty cells may still have some signal due to RNA in the media.
In rare cases a bead may be combined with 2 or more cells.
This is particularly problematic when the 2+ cells are all from the same cell type.</p><p>10X Genomics provides their own software for demultiplexing and identifying cells (<code>Cell Ranger</code>).
This software does demultiplexes reads to the cell level.
Cells are called using total UMI count:</p><blockquote><p>Cell Ranger v3 (<a href=https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/algorithms/overview>website</a>)</p><ol><li>Uses a cutoff based on total UMI counts of each barcode to identify cells. This step identifies the primary mode of high RNA content cells.</li><li>Then the algorithm uses the RNA profile of each remaining barcode to determine if it is an “empty&rdquo; or a cell containing partition. This second step captures low RNA content cells whose total UMI counts may be similar to empty GEMs.</li></ol><p>Cell Ranger v2 only uses step (1).</p></blockquote><p>After cell selection <code>Cell Ranger</code> aligns reads using <code>STAR</code>.
It counts the number of reads per gene and provides summary statistics.</p><h3 id=cell-ranger-v2-1-1>Cell Ranger (v2.1.1)</h3><p><a href=https://github.com/jfear/larval_gonad/blob/5f8f4569ea253ab96d497055e7ae6ebb4c82a744/scrnaseq-wf/Snakefile#L96-L121 target=_blank class="btn btn-default"></p><pre><code>Code
</code></pre><p></a></p><p>10X Genomics suggests a capture rate of ~50% of cells loaded. We expect to capture at least 3,000 cells for each replicate.
Using v2.1.1 of <code>Cell Ranger</code> with default settings we capture ~500 cells per replicate (below).
The barcode rank plots indicate a long sloped decrease (left to right) with a knee appearing around 100 UMI.
We think <code>Cell Ranger</code> is under calling cells.
10X reported on their website that the v2 of their cell calling algorithm struggles when there is a high diversity of RNA content.
In our data we expect a high diversity of RNA content because we are looking at stem cells, germ cells, and somatic cells.
Therefore, we decided to try forcing the algorithm to call additional cells.</p><h4 id=testis-replicate-1>Testis Replicate 1</h4><div class=expand><div class=expand-label style=cursor:pointer onclick="$h=$(this);$h.next('div').slideToggle(100,function(){$h.children('i').attr('class',function(){return $h.next('div').is(':visible')?'fas fa-chevron-down':'fas fa-chevron-right';});});"><i style=font-size:x-small class="fas fa-chevron-right"></i><span>click to expand</span></div><div class=expand-content style=display:none><figure><img src=cellranger2_testis1_summary.png></figure></div></div><h4 id=testis-replicate-2>Testis Replicate 2</h4><div class=expand><div class=expand-label style=cursor:pointer onclick="$h=$(this);$h.next('div').slideToggle(100,function(){$h.children('i').attr('class',function(){return $h.next('div').is(':visible')?'fas fa-chevron-down':'fas fa-chevron-right';});});"><i style=font-size:x-small class="fas fa-chevron-right"></i><span>click to expand</span></div><div class=expand-content style=display:none><figure><img src=cellranger2_testis2_summary.png></figure></div></div><h4 id=testis-replicate-3>Testis Replicate 3</h4><div class=expand><div class=expand-label style=cursor:pointer onclick="$h=$(this);$h.next('div').slideToggle(100,function(){$h.children('i').attr('class',function(){return $h.next('div').is(':visible')?'fas fa-chevron-down':'fas fa-chevron-right';});});"><i style=font-size:x-small class="fas fa-chevron-right"></i><span>click to expand</span></div><div class=expand-content style=display:none><figure><img src=cellranger2_testis3_summary.png></figure></div></div><h3 id=cell-ranger-v2-1-1-force-cells>Cell Ranger (v2.1.1; <code>--force-cells</code>)</h3><p><a href=https://github.com/jfear/larval_gonad/blob/5f8f4569ea253ab96d497055e7ae6ebb4c82a744/scrnaseq-wf/Snakefile#L123-L151 target=_blank class="btn btn-default"></p><pre><code>Code
</code></pre><p></a></p><p><code>Cell Ranger</code> provides a way to force a certain number of cells to be called.
We ran <code>Cell Ranger</code> using this option to force the calling of 3,000 cells for replicates 1 and 2 where we loaded ~6,000 cells.
We loaded ~16,000 cells in replicate 3 so we force 8,000 cells to be called.
Again looking at the barcode rank plots we see that cells were called all the way to the middle of the linear portion of the distribution (~ ≥1,000 UMI).
The median total UMI per cell ranged from 1,091-5,429.
The median number of genes per cell ranged from 546-1,120.
Replicates 2 and 3 had ~20,000,000 more reads than replicate 1.</p><h4 id=testis-replicate-1-1>Testis Replicate 1</h4><div class=expand><div class=expand-label style=cursor:pointer onclick="$h=$(this);$h.next('div').slideToggle(100,function(){$h.children('i').attr('class',function(){return $h.next('div').is(':visible')?'fas fa-chevron-down':'fas fa-chevron-right';});});"><i style=font-size:x-small class="fas fa-chevron-right"></i><span>click to expand</span></div><div class=expand-content style=display:none><figure><img src=cellranger2_testis1_force_summary.png></figure></div></div><h4 id=testis-replicate-2-1>Testis Replicate 2</h4><div class=expand><div class=expand-label style=cursor:pointer onclick="$h=$(this);$h.next('div').slideToggle(100,function(){$h.children('i').attr('class',function(){return $h.next('div').is(':visible')?'fas fa-chevron-down':'fas fa-chevron-right';});});"><i style=font-size:x-small class="fas fa-chevron-right"></i><span>click to expand</span></div><div class=expand-content style=display:none><figure><img src=cellranger2_testis2_force_summary.png></figure></div></div><h4 id=testis-replicate-3-1>Testis Replicate 3</h4><div class=expand><div class=expand-label style=cursor:pointer onclick="$h=$(this);$h.next('div').slideToggle(100,function(){$h.children('i').attr('class',function(){return $h.next('div').is(':visible')?'fas fa-chevron-down':'fas fa-chevron-right';});});"><i style=font-size:x-small class="fas fa-chevron-right"></i><span>click to expand</span></div><div class=expand-content style=display:none><figure><img src=cellranger2_testis3_force_summary.png></figure></div></div><h3 id=cell-ranger-v3-0-2>Cell Ranger (v3.0.2)</h3><p><a href=https://github.com/jfear/larval_gonad/blob/5f8f4569ea253ab96d497055e7ae6ebb4c82a744/scrnaseq-wf/Snakefile#L123-L151 target=_blank class="btn btn-default"></p><pre><code>Code
</code></pre><p></a></p><p>10X recently released v3 of their <code>Cell Ranger</code> software.
In this version they boast a better cell calling with low RNA content.</p><h4 id=testis-replicate-1-2>Testis Replicate 1</h4><div class=expand><div class=expand-label style=cursor:pointer onclick="$h=$(this);$h.next('div').slideToggle(100,function(){$h.children('i').attr('class',function(){return $h.next('div').is(':visible')?'fas fa-chevron-down':'fas fa-chevron-right';});});"><i style=font-size:x-small class="fas fa-chevron-right"></i><span>click to expand</span></div><div class=expand-content style=display:none><figure><img src=cellranger3_testis1_summary.png></figure></div></div><h4 id=testis-replicate-2-2>Testis Replicate 2</h4><div class=expand><div class=expand-label style=cursor:pointer onclick="$h=$(this);$h.next('div').slideToggle(100,function(){$h.children('i').attr('class',function(){return $h.next('div').is(':visible')?'fas fa-chevron-down':'fas fa-chevron-right';});});"><i style=font-size:x-small class="fas fa-chevron-right"></i><span>click to expand</span></div><div class=expand-content style=display:none><figure><img src=cellranger3_testis2_summary.png></figure></div></div><h4 id=testis-replicate-3-2>Testis Replicate 3</h4><div class=expand><div class=expand-label style=cursor:pointer onclick="$h=$(this);$h.next('div').slideToggle(100,function(){$h.children('i').attr('class',function(){return $h.next('div').is(':visible')?'fas fa-chevron-down':'fas fa-chevron-right';});});"><i style=font-size:x-small class="fas fa-chevron-right"></i><span>click to expand</span></div><div class=expand-content style=display:none><figure><img src=cellranger3_testis3_summary.png></figure></div></div></article><div class="book-footer justify-between"><div><a href=https://github.com/jfear/larval_gonad_docs/commit/504880934e55a38b0ed2a9ce3749bdd5245bab13 title="Last modified May 28, 2019 by Justin M. Fear" target=_blank rel=noopener><img src=/svg/calendar.svg alt=Changed> May 28, 2019</a></div><div><a href=https://github.com/jfear/larval_gonad_docs/edit/master/content/docs/scrnaseq_prep/cell_selection/_index.md target=_blank rel=noopener><img src=/svg/edit.svg alt=Edit> Edit this page</a></div></div></div><aside class="book-toc level-4 fixed"><nav id=TableOfContents><ul><li><ul><li><a href=#barcode-rank-plots>Barcode Rank Plots</a><ul><li><a href=#cell-ranger-v2-1-1>Cell Ranger (v2.1.1)</a><ul><li><a href=#testis-replicate-1>Testis Replicate 1</a></li><li><a href=#testis-replicate-2>Testis Replicate 2</a></li><li><a href=#testis-replicate-3>Testis Replicate 3</a></li></ul></li><li><a href=#cell-ranger-v2-1-1-force-cells>Cell Ranger (v2.1.1; <code>--force-cells</code>)</a><ul><li><a href=#testis-replicate-1-1>Testis Replicate 1</a></li><li><a href=#testis-replicate-2-1>Testis Replicate 2</a></li><li><a href=#testis-replicate-3-1>Testis Replicate 3</a></li></ul></li><li><a href=#cell-ranger-v3-0-2>Cell Ranger (v3.0.2)</a><ul><li><a href=#testis-replicate-1-2>Testis Replicate 1</a></li><li><a href=#testis-replicate-2-2>Testis Replicate 2</a></li><li><a href=#testis-replicate-3-2>Testis Replicate 3</a></li></ul></li></ul></li></ul></li></ul></nav></aside></main></body></html>