<!doctype html><html lang=en><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><title>Cell Selection
| Larval Testis X Inactivation</title><link rel=stylesheet href=/larval_gonad_docs/book.min.e0adbac1e0fc6fdc3ef34548c686f58a7e1ba9fb1adf5c9591aa4b7df3653f22.css><link rel=icon href=/larval_gonad_docs/favicon.png type=image/x-icon><link rel=alternate type=application/rss+xml href=https://jfear.github.io/larval_gonad_docs/docs/scrnaseq_prep/cell_selection/index.xml title="Larval Testis X Inactivation"></head><body><input type=checkbox style=display:none id=menu-control><main class="flex container"><aside class="book-menu fixed"><nav><h2 class=book-brand><a href=https://jfear.github.io/larval_gonad_docs>Larval Testis X Inactivation</a></h2><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/>Data Prep and Quality Control</a><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cell_selection/ class=active>Cell Selection</a><ul></ul></li><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cluster_selection/>Selecting Clustering Algorithm</a><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cluster_selection/seurat2/testis1/>Seurat2 Testis 1 Clustering</a></li></ul></li></ul></li><li><a href=/larval_gonad_docs/docs/clustering/>Clustering</a><ul><li><a href=/larval_gonad_docs/docs/clustering/combined_n3/>Integrated Clustering (n = 3)</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis1/>Testis 1 Individual Clustering</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis2/>Testis 2 Individual Clustering</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis3/>Testis 3 Individual Clustering</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis4/>Testis 4 Individual Clustering</a></li></ul></li><li><a href=/larval_gonad_docs/docs/intronless_genes/>Intronless Gene Expression</a><ul></ul></li><li><a href=/larval_gonad_docs/docs/gene_movement/>Neo X Gene Movement</a><ul></ul></li></ul></nav><script>(function(){var menu=document.querySelector('aside.book-menu nav')
addEventListener('beforeunload',function(event){localStorage.setItem('menu.scrollTop',menu.scrollTop)});menu.scrollTop=localStorage.getItem('menu.scrollTop')})()</script></aside><div class=book-page><header class="align-center justify-between book-header"><label for=menu-control><img src=/larval_gonad_docs/svg/menu.svg alt=Menu></label>
<strong>Cell Selection</strong></header><article class=markdown><h2 id=cell-selection>Cell Selection</h2><h3 id=tl-dr>TL;DR</h3><p><strong>Problem:</strong> Identify individual cells from scRNA-Seq data.</p><p>The first step in scRNA-Seq analysis is to identify which reads come from individual cells.
This requires removing empty GEMs (<strong>Figure 1</strong>), heterotypic and homotypic multiplets, and cells with low expression diversity.
I selected the following as putative cells for clustering and downstream analysis (<strong>Table 1</strong>).
Replicates 3 and 4 are puzzling, with having more captured cells than expected.
Either I am over-calling cells or more cells were loaded than thought.</p><p><strong>Table 1. Final cell selection counts.</strong></p><table><thead><tr><th>Replicate</th><th>Approx. Num. Cells Loaded</th><th>Expected Num. Cells Captured</th><th>Num. of Cells Selected</th><th>Num. of Reads</th></tr></thead><tbody><tr><td>Testis 1</td><td>6,000</td><td>3,000</td><td>2,710</td><td>119,932,060</td></tr><tr><td>Testis 2</td><td>6,000</td><td>3,000</td><td>4,226</td><td>148,921,414</td></tr><tr><td>Testis 3</td><td>12,000</td><td>8,000</td><td>11,946</td><td>131,590,010</td></tr><tr><td>Testis 4</td><td>12,000</td><td>8,000</td><td>14,697</td><td>159,678,110</td></tr></tbody></table><h3 id=overview>Overview</h3><figure><img src="https://cdn.technologynetworks.com/tn/images/thumbs/webp/640_360/10x-genomics-extends-their-application-portfolio-305346.webp?v=9720678" alt="Figure 1. The 10X Library Preparation consists of mixing cells with gel beads (GEMs). The goal of this process is to get a single cell associated with a single GEM." width=80%><figcaption><p><strong>Figure 1. The 10X Library Preparation consists of mixing cells with gel beads (GEMs).</strong> The goal of this process is to get a single cell associated with a single GEM.</p></figcaption></figure><p>10X genomics is a droplet based method where cells are mixed with gel beads (GEMs) that contain indices and all materials needed for the first strand cDNA synthesis (<strong>Figure 1</strong>).
Cells are loaded with GEMs using a Poisson distribution which results in most GEMs being empty.
The 10X genomics kit that we are using has 737,280 GEMs per sample.
Oligonucleotides attached to each GEM contain a cell index and a molecule index.
I want to only use GEMs capturing a single cell, so I need to remove empty GEMs and GEMs that captured more than 1 cell.</p><p>We can model total unique molecule expression (UMI) to separate GEMs that contain ≥1 cell from empty GEMs.
This will account for any ambient RNA in the media.
10X Genomics provides their own software for calling cells (<a href=https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/algorithms/overview><code>cell ranger</code></a>).
Recently, 10X genomics released version 3 of <code>cell ranger</code>, which boasts improved cell calling for low RNA content cells.
Here I evaluate <code>cell ranger</code> <code>v2</code> and <code>v3</code> as well as a 3rd party tool <code>DropLetUtils</code>.</p><p>In addition to removing GEMs with 0 cells, we also want to remove GEMs with ≥2 cells (a.k.a multiplets).
There are two types of multiplets that can occur:</p><ul><li>Heterotypic: ≥2 cells from different cell types.</li><li>Homeotypic: ≥2 cells from the same cell type</li></ul><p>Heterotypic multiplets can be identified <em>in silico</em>.
Essentially I create synthetic doublets by mixing two or more cells from different cell type clusters.
I then re-cluster cells + synthetic cells.
Cells that cluster with synthetic doublets are likely to be heterotypic multiplets and are removed.
There are several tools available to do this analysis, I decided to use <code>scrublet</code> because it works with raw 10X data.</p><p>We do expect biases when it comes to the types of multiplets that we capture.
We do not expect to capture doublets of large cells or large aggregates because there is a filtration step (35 μm) during sample prep.
In particular, fused later staged primary spermatocytes are too large (~50 μm) to pass through the filter.
While multiplets from smaller somatic cells, spermatogonia, or early stage primary spermatocytes may easily pass through the filter.
Sharvani does see evidence for multiple small cells stuck together after filtration.
She is working on getting an estimate of the frequency of multiplets to give better insight of how big of a problem this may be.</p><h3 id=results>Results</h3><p><strong>NOTE:</strong> Below I display results for each replicate on different tabs.
Click the replicate name to view results.
Complete descriptions of the data are included with <strong>Testis 1</strong>.</p><h4 id=empty-gem-identification>Empty GEM Identification</h4><p>To identify empty GEMs I used multiple software and parameterizations.</p><p><strong><code>cellranger-wf</code></strong> uses cell ranger v2 with default settings.
This workflow calls the fewest number of cells (~500 cells / rep).
According to 10X Genomics we expect to capture ~50% of loaded cells, but this method gives ≤8%.
This method is known not to perform well when there is a large dynamic range of RNA content.
Our samples are from the <em>Drosophila</em> testis which is known to contain nearly quiescent cells and extremely high RNA content cells.
Therefore, we think this method is under calling captured cells.</p><p><strong><code>cellranger-force-wf</code></strong> uses cell ranger v2 with the <code>--force-cells</code> option.
When samples contain a high dynamic range of RNA content, 10X Genomics suggests to use the <code>--force-cells</code>.
Essentially this method orders cells according to total UMI content and then keeps the highest <code>N</code> cells.
We specified our forced settings to capture 50% of loaded cells, but this is <em>ad hoc</em> and likely over calls captured cells.</p><p><strong><code>cellranger3-wf</code></strong> uses cell ranger v3 with default settings.
Recently, 10X Genomics released version 3 of cell ranger.
This version adds an additional step to the cell calling algorithm which improves calling of low RNA content cells.
Interestingly, this method called more cells than <code>cellranger-force-wf</code> in all replicates expect <strong>Testis 1</strong>.</p><p><strong><code>droputils</code></strong> uses DropLetUtils v3.0.1 with default settings.
DropLetUtils is an R packaged with several algorithms for scRNA-Seq.
I used the <code>emptydrops</code> function which models UMI to classify empty GEMs.
This method performs very different for each replicate.
This method seems to over call cells in <strong>Testis 1</strong>, <strong>Testis 3</strong>, and <strong>Testis 4</strong> while performing similarly to <code>cell ranger</code> on <strong>Testis 2</strong>.</p><p>Without a ground truth data set it is impossible to know which method is best approximating real cell calls.
Consensus is often used in the absence of truth.
However, sense all of these methods use total UMI in their models, the 4-way consensus would be the same as the most conservative cell calls (<code>cellranger-wf</code>).
Given the known limitation of <code>cellranger-wf</code> and the <em>ad hoc</em> nature of <code>cellranger-force-wf</code> I decided to use the consensus of <code>cellranger3-wf</code> and <code>droputils</code>.</p><p><strong>Table 2. Cell count after removing empty GEMs.</strong></p><table><thead><tr><th>Replicate</th><th><code>cellranger-wf</code></th><th><code>cellranger-force-wf</code></th><th><code>cellranger3-wf</code></th><th><code>droputils</code></th><th>2-way Consensus<sup>‡</sup></th></tr></thead><tbody><tr><td>Testis 1</td><td>483</td><td>3,000</td><td>2,826</td><td>13,884</td><td>2,790</td></tr><tr><td>Testis 2</td><td>550</td><td>3,000</td><td>6,385</td><td>5,765</td><td>4,801</td></tr><tr><td>Testis 3</td><td>423</td><td>8,000</td><td>12,485</td><td>39,872</td><td>12,515</td></tr><tr><td>Testis 4</td><td>349</td><td>8,000</td><td>15,033</td><td>34,761</td><td>15,033</td></tr></tbody></table><p><sup>‡</sup>Intersection of <code>cellranger3-wf</code> and <code>droputils</code>.</p><div class=book-tabs><input type=radio name="tabs-Cell Selection By Sample" style=display:none id="tabs-Cell Selection By Sample-0" checked>
<label for="tabs-Cell Selection By Sample-0">Testis 1</label><div class="book-tabs-content markdown-inner"><h5 id=cellranger-wf-v2-1-1-defaults><code>cellranger-wf</code> (v2.1.1; defaults)</h5><figure><img src=cellranger2_testis1_summary.png width=100%></figure><p>Summary statistics and barcode rank plot provided by <code>cell ranger</code>.
Things to pay attention to:</p><ul><li><p>The three numbers in green at the top (<em>Estimated Number of Cells</em>, <em>Mean Reads per Cell</em>, and <em>Median Genes per Cell</em>).
We are looking for a balance between these numbers.
As cells with lower UMI are added, the <em>Mean Reads per Cell</em> and <em>Median Genes per Cell</em> will decrease.
We do not want these numbers to get too low as the added cells will not have enough signal.</p></li><li><p>The <em>Fraction of Reads in Cells</em> (2nd number below plot).
We expect that most reads should be in a cell and not in empty GEMs.</p></li><li><p>The percent of <em>Reads Mapped to Genome</em>.
This number will be the same across cell ranger runs, but indicates how good our samples are.
<strong>Testis 1</strong> has the lowest mapping rate of 73%.
This is considered moderate to low quality in Bulk RNA-Seq studies where we typically aim for ≥90% mapping.</p></li><li><p>The Barcode Rank Plot.
This plot shows the Log Total UMI Count (Y-axis) vs the Log Rank Ordered Cell Count (X-axis).
Light gray points are empty GEMs and colored points are considered non-empty cells.
The goal of cell selection is to find the cutoff along this curve.</p></li></ul><h5 id=cellranger-force-wf-v2-1-1-force><code>cellranger-force-wf</code> (v2.1.1; force)</h5><p>Note the <em>Fraction Reads in Cells</em> is closer to the &gt;70%.</p><figure><img src=cellranger2_testis1_force_summary.png width=100%></figure><h5 id=cellranger3-wf-v3-0-1-defaults>`cellranger3-wf (v3.0.1; defaults)</h5><p>These results are very similar to <code>cellranger-force-wf</code>.</p><figure><img src=cellranger3_testis1_summary.png width=100%></figure><h5 id=droputils-dropletutils-v1-2-1><code>droputils</code> (DropletUtils v1.2.1)</h5><p>Finally, I wanted to use an external tool for cell selection.
The most popular tool I founds was an R package called <code>dropletUtils</code>.</p><blockquote><p>Lun ATL, Riesenfeld S, Andrews T, Dao T, Gomes T, participants in the 1st
Human Cell Atlas Jamboree, Marioni JC (2019). “EmptyDrops: distinguishing
cells from empty droplets in droplet-based single-cell RNA sequencing data.”
Genome Biol., 20, 63. doi: 10.1186/s13059-019-1662-y.</p></blockquote><p>This tool also builds a classification model based on UMI content.
I do not have all of the summary stats for these cell calls, but I proved the barcode rank plot.
The tool estimates the knee (blue line) and the inflection point (green line) automatically.
Putative empty GEMs are in light gray and non-empty cells are in black.
This method always calls way more cells than <code>cell ranger</code> methods.</p><figure><img src=dropletutils_testis1_bc_rank.png width=70%></figure><h5 id=pairwise-similarity-jaccard>Pairwise Similarity (Jaccard)</h5><p>In this section I look at the similarity of calls between the different methods using the pairwise Jacaard similarity score.
The <code>cellranger-foce-wf</code> and the <code>cellranger3-wf</code> always have the highest similarity.
The <code>droputils</code> and the <code>cellranger-wf</code> always have the lowest similarity.</p><table><thead><tr><th>method</th><th>cellranger-wf</th><th>cellranger-force-wf</th><th>cellranger3-wf</th><th>droputils</th></tr></thead><tbody><tr><td>cellranger-wf</td><td>1</td><td>0.8219</td><td>0.8342</td><td>0.0537</td></tr><tr><td>cellranger-force-wf</td><td>0.8219</td><td>1</td><td>0.9771</td><td>0.1919</td></tr><tr><td>cellranger3-wf</td><td>0.8342</td><td>0.9771</td><td>1</td><td>0.2146</td></tr><tr><td>droputils</td><td>0.0537</td><td>0.1919</td><td>0.2146</td><td>1</td></tr></tbody></table><h5 id=consensus>Consensus</h5><p>I provide a summary for the 3-way consensus (<code>cellranger-force-wf</code>, <code>cellranger3-wf</code>, <code>droputils</code>) and the 2-way consensus (<code>cellranger3-wf</code> and <code>droputils</code>).</p><ul><li>Number of cells with 3-way consensus: 2,717</li><li>Number of cells with 2-way consensus: 2,790</li><li>Number of different calls between consensus with 3 vs 2 measures: 73</li><li>Jaccard similarity of consensus with 3 vs 2 measures: 0.9948479074034865</li></ul><figure><img src=testis1_combined_upset.svg width=100%></figure><p>This is an UpSet plot, it is like a high dimensional Venn Diagram.
Each bar represents the number of cells for a given set.
The set is represented below as black dots.
For example the first bar is the 4-way consensus (intersection of all 4 methods).</p></div><input type=radio name="tabs-Cell Selection By Sample" style=display:none id="tabs-Cell Selection By Sample-1">
<label for="tabs-Cell Selection By Sample-1">Testis 2</label><div class="book-tabs-content markdown-inner"><h5 id=cellranger-wf-v2-1-1-defaults><code>cellranger-wf</code> (v2.1.1; defaults)</h5><figure><img src=cellranger2_testis2_summary.png width=100%></figure><h5 id=cellranger-force-wf-v2-1-1-force><code>cellranger-force-wf</code> (v2.1.1; force)</h5><figure><img src=cellranger2_testis2_force_summary.png width=100%></figure><h5 id=cellranger3-wf-v3-0-1-defaults><code>cellranger3-wf</code> (v3.0.1; defaults)</h5><figure><img src=cellranger3_testis2_summary.png width=100%></figure><h5 id=droputils-dropletutils-v1-2-1><code>droputils</code> (DropletUtils v1.2.1)</h5><figure><img src=dropletutils_testis2_bc_rank.png width=70%></figure><h5 id=pairwise-similarity-jaccard>Pairwise Similarity (Jaccard)</h5><table><thead><tr><th>method</th><th>cellranger-wf</th><th>cellranger-force-wf</th><th>cellranger3-wf</th><th>droputils</th></tr></thead><tbody><tr><td>cellranger-wf</td><td>1</td><td>0.6678</td><td>0.2083</td><td>0.2933</td></tr><tr><td>cellranger-force-wf</td><td>0.6678</td><td>1</td><td>0.5335</td><td>0.6081</td></tr><tr><td>cellranger3-wf</td><td>0.2083</td><td>0.5335</td><td>1</td><td>0.6539</td></tr><tr><td>droputils</td><td>0.2933</td><td>0.6081</td><td>0.6539</td><td>1</td></tr></tbody></table><h5 id=consensus>Consensus</h5><ul><li>Number of cells with 3-way consensus: 2,936</li><li>Number of cells with 2-way consensus: 4,801</li><li>Number of different calls between consensus with 3 vs 2 measures: 1,865</li><li>Jaccard similarity of consensus with 3 vs 2 measures: 0.7473584394473043</li></ul><figure><img src=testis2_combined_upset.svg width=100%></figure></div><input type=radio name="tabs-Cell Selection By Sample" style=display:none id="tabs-Cell Selection By Sample-2">
<label for="tabs-Cell Selection By Sample-2">Testis 3</label><div class="book-tabs-content markdown-inner"><h5 id=cellranger-wf-v2-1-1-defaults><code>cellranger-wf</code> (v2.1.1; defaults)</h5><figure><img src=cellranger2_testis3_summary.png width=100%></figure><h5 id=cellranger-force-wf-v2-1-1-force><code>cellranger-force-wf</code> (v2.1.1; force)</h5><figure><img src=cellranger2_testis3_force_summary.png width=100%></figure><h5 id=cellranger3-wf-v3-0-1-defaults><code>cellranger3-wf</code> (v3.0.1; defaults)</h5><figure><img src=cellranger3_testis3_summary.png width=100%></figure><h5 id=droputils-dropletutils-v1-2-1><code>droputils</code> (DropletUtils v1.2.1)</h5><figure><img src=dropletutils_testis3_bc_rank.png width=70%></figure><h5 id=pairwise-similarity-jaccard>Pairwise Similarity (Jaccard)</h5><table><thead><tr><th>method</th><th>cellranger-wf</th><th>cellranger-force-wf</th><th>cellranger3-wf</th><th>droputils</th></tr></thead><tbody><tr><td>cellranger-wf</td><td>1</td><td>0.81</td><td>0.6968</td><td>0.0107</td></tr><tr><td>cellranger-force-wf</td><td>0.81</td><td>1</td><td>0.8489</td><td>0.2006</td></tr><tr><td>cellranger3-wf</td><td>0.6968</td><td>0.8489</td><td>1</td><td>0.3139</td></tr><tr><td>droputils</td><td>0.0107</td><td>0.2006</td><td>0.3139</td><td>1</td></tr></tbody></table><h5 id=consensus>Consensus</h5><ul><li>Number of cells with 3-way consensus: 7,245</li><li>Number of cells with 2-way consensus: 12,515</li><li>Number of different calls between consensus with 3 vs 2 measures: 5,270</li><li>Jaccard similarity of consensus with 3 vs 2 measures: 0.8678270465489567</li></ul><figure><img src=testis3_combined_upset.svg width=100%></figure></div><input type=radio name="tabs-Cell Selection By Sample" style=display:none id="tabs-Cell Selection By Sample-3">
<label for="tabs-Cell Selection By Sample-3">Testis 4</label><div class="book-tabs-content markdown-inner"><h5 id=cellranger-wf-v2-1-1-defaults><code>cellranger-wf</code> (v2.1.1; defaults)</h5><figure><img src=cellranger2_testis4_summary.png width=100%></figure><h5 id=cellranger-force-wf-v2-1-1-force><code>cellranger-force-wf</code> (v2.1.1; force)</h5><figure><img src=cellranger2_testis4_force_summary.png width=100%></figure><h5 id=cellranger3-wf-v3-0-1-defaults><code>cellranger3-wf</code> (v3.0.1; defaults)</h5><figure><img src=cellranger3_testis4_summary.png width=100%></figure><h5 id=droputils-dropletutils-v1-2-1><code>droputils</code> (DropletUtils v1.2.1)</h5><figure><img src=dropletutils_testis4_bc_rank.png width=70%></figure><h5 id=pairwise-similarity-jaccard>Pairwise Similarity (Jaccard)</h5><table><thead><tr><th>method</th><th>cellranger-wf</th><th>cellranger-force-wf</th><th>cellranger3-wf</th><th>droputils</th></tr></thead><tbody><tr><td>cellranger-wf</td><td>1</td><td>0.9237</td><td>0.5776</td><td>0.01</td></tr><tr><td>cellranger-force-wf</td><td>0.9237</td><td>1</td><td>0.6538</td><td>0.0863</td></tr><tr><td>cellranger3-wf</td><td>0.5776</td><td>0.6538</td><td>1</td><td>0.4325</td></tr><tr><td>droputils</td><td>0.01</td><td>0.0863</td><td>0.4325</td><td>1</td></tr></tbody></table><h5 id=consensus>Consensus</h5><ul><li>Number of cells with 3-way consensus: 3,000</li><li>Number of cells with 2-way consensus: 15,033</li><li>Number of different calls between consensus with 3 vs 2 measures: 12,033</li><li>Jaccard similarity of consensus with 3 vs 2 measures: 0.6538361957366013</li></ul><figure><img src=testis4_combined_upset.svg width=100%></figure></div></div><p></p><h4 id=heterotypic-doublet-detection>Heterotypic Doublet Detection</h4><p>Next I wanted to identify cells that are heterotypic doublets.
There are several software packages available, but all use essentially the same algorithm.</p><ol><li>Cells are clustered into individual cells types</li><li>Cells from different clusters are combined <em>in silico</em> to form synthetic doublets.</li><li>Cells + synthetic doublets are mixed and re-clustered.</li><li>Cells that cluster with the synthetic doublets are likely heterotypic multiplets.</li></ol><p>Again outputs are displayed below in different tabs for different samples.
On the <strong>Testis 1</strong> I added additional text describing what results are provided.</p><p>I am using cells from the 2-way consensus above.</p><p><strong>Table 3. Summary of homotypic multiplets.</strong></p><table><thead><tr><th>Replicate</th><th>Num Homotypic Multiplets</th></tr></thead><tbody><tr><td>Testis 1</td><td>48</td></tr><tr><td>Testis 2</td><td>45</td></tr><tr><td>Testis 3</td><td>169</td></tr><tr><td>Testis 4</td><td>60</td></tr></tbody></table><div class=book-tabs><input type=radio name="tabs-Doublet Detection By Sample" style=display:none id="tabs-Doublet Detection By Sample-0" checked>
<label for="tabs-Doublet Detection By Sample-0">Testis 1</label><div class="book-tabs-content markdown-inner"><h5 id=simulation-cutoff>Simulation Cutoff</h5><p><code>scrublet</code> provides this diagnostic histogram to help identify the threshold for what is called a doublet.
In the right panel are the simulated doublets.
It is expected to be a bimodal distribution with doublets being on the right and singlets (or homeotypic doublets) being on the left.
Our data does not provide a clear modality, this is probably due to the fact we have two lineage and a large dynamic range of RNA content.
I set the threshold (t = 0.25) to be the same for all replicates.</p><figure><img src=testis1_scrublet_histogram.png width=100%></figure><h5 id=projects-of-doublets-n-48>Projects of Doublets (n = 48)</h5><p>This plot shows the UMAP projection colored by doublet calls.
In <strong>Testis 1</strong> we can see these intermediate cell types being labeled as doublets.
This is not true for the other replicates.</p><figure><img src=testis1_scrublet_umap.png width=100%></figure></div><input type=radio name="tabs-Doublet Detection By Sample" style=display:none id="tabs-Doublet Detection By Sample-1">
<label for="tabs-Doublet Detection By Sample-1">Testis 2</label><div class="book-tabs-content markdown-inner"><h5 id=simulation-cutoff>Simulation Cutoff</h5><figure><img src=testis2_scrublet_histogram.png width=100%></figure><h5 id=projects-of-doublets-n-45>Projects of Doublets (n = 45)</h5><figure><img src=testis2_scrublet_umap.png width=100%></figure></div><input type=radio name="tabs-Doublet Detection By Sample" style=display:none id="tabs-Doublet Detection By Sample-2">
<label for="tabs-Doublet Detection By Sample-2">Testis 3</label><div class="book-tabs-content markdown-inner"><h5 id=simulation-cutoff>Simulation Cutoff</h5><figure><img src=testis3_scrublet_histogram.png width=100%></figure><h5 id=projects-of-doublets-n-169>Projects of Doublets (n = 169)</h5><figure><img src=testis3_scrublet_umap.png width=100%></figure></div><input type=radio name="tabs-Doublet Detection By Sample" style=display:none id="tabs-Doublet Detection By Sample-3">
<label for="tabs-Doublet Detection By Sample-3">Testis 4</label><div class="book-tabs-content markdown-inner"><h5 id=simulation-cutoff>Simulation Cutoff</h5><figure><img src=testis4_scrublet_histogram.png width=100%></figure><h5 id=projects-of-doublets-n-60>Projects of Doublets (n = 60)</h5><figure><img src=testis4_scrublet_umap.png width=100%></figure></div></div><p></p><h4 id=homotypic-doublet-and-low-complexity-detection>Homotypic Doublet and Low Complexity Detection</h4><p>Finally, I attempt to remove homotypic cells using a high expression cutoff.
At the same time I am checking if additional low expression cutoffs would be useful, and if I should remove cells with high mitochondrial or rRNA expression.
In the absence of truth, setting these thresholds is <em>ad hoc</em>.
To get a better understanding of the value of these different criteria I use a grid search.
For each criteria (low gene cutoff, high gene cutoff, percent mitochondiral reads, percent rRNA reads) I selected 4 levels.
I ran Seurat (v2) clustering on the different combinations in these same (12 dimensions), aiming to get 12-13 clusters.
I used the Adjusted Rand Index to look at similarity of cluster calls for each pairwise comparison.</p><p>In general, filtering by the percent mitochondrial or percent ribosomal had no affect on clustering.
Low end cutoffs of 200 or 500 expressed genes behaved very similarly, while increasing this to 1,000 gene changes clustering outcome.
In <strong>Testis 1</strong> there was little difference using a high end cutoff.
However, in other samples any high cutoff created different clusters than using no high end cutoff.
I am interpreting this as the most extreme high expressing cells can affect clustering.</p><h5 id=cutoff-criteria>Cutoff Criteria</h5><ul><li><strong>NO</strong> Mitochondrial criteria</li><li><strong>NO</strong> rRNA criteria</li><li>Remove cells <strong>≤200 expressed genes</strong>.</li><li>Remove cells <strong>≥5,000 expressed genes</strong>.</li></ul><div class=book-tabs><input type=radio name="tabs-Gene Level Filters" style=display:none id="tabs-Gene Level Filters-0" checked>
<label for="tabs-Gene Level Filters-0">Testis 1</label><div class="book-tabs-content markdown-inner"><h5 id=distribution-of-cells-based-on-gene-level-counts>Distribution of cells based on gene level counts</h5><figure><img src=./testis1_vln.png width=80%></figure><p>Cell distributions for:</p><ul><li>nGene: The total number genes expressed (&gt;0). Cells at the high end of this distribution are possibly homotypic doublets.</li><li>nUMI: The total UMI count per cell. Cells at the high end of this distribution are possibly homotypic doublets.</li><li>percent.mito: The percent of reads mapping to the mitochondria. Cells at the high end of this distribution are possibly dying.</li><li>percent.ribo: The percent reads mapping to the rRNA. Cells at the high end of this distribution possibly had poor poly(A) selection.</li></ul><h5 id=relationship-of-umi-vs-other-measures>Relationship of UMI vs other measures</h5><figure><img src=./testis1_scatter.png width=80%></figure><p>As expeted the nGene and nUMI are highly correlated.
However, cells high in percent mitochondrial or ribosomal reads tend to be in the low to middle of the UMI distribution.</p><h5 id=grid-search-summary>Grid search summary</h5><figure><a href=./testis1_heatmap_grid.svg target=_blank><img src=./testis1_heatmap_grid.svg alt="click image to enlarge" width=100%></a><figcaption><p>click image to enlarge</p></figcaption></figure><p>The similarity matrix of <a href=https://scikit-learn.org/stable/modules/clustering.html#adjusted-rand-score>Adjusted Rand Indexes</a>.
A score of 1 indicates cluster calls were identical.</p></div><input type=radio name="tabs-Gene Level Filters" style=display:none id="tabs-Gene Level Filters-1">
<label for="tabs-Gene Level Filters-1">Testis 2</label><div class="book-tabs-content markdown-inner"><h5 id=distribution-of-cells-based-on-gene-level-counts>Distribution of cells based on gene level counts</h5><figure><img src=./testis2_vln.png width=80%></figure><h5 id=relationship-of-umi-vs-other-measures>Relationship of UMI vs other measures</h5><figure><img src=./testis2_scatter.png width=80%></figure><h5 id=grid-search-summary>Grid search summary</h5><figure><a href=./testis2_heatmap_grid.svg target=_blank><img src=./testis2_heatmap_grid.svg alt="click image to enlarge" width=100%></a><figcaption><p>click image to enlarge</p></figcaption></figure></div><input type=radio name="tabs-Gene Level Filters" style=display:none id="tabs-Gene Level Filters-2">
<label for="tabs-Gene Level Filters-2">Testis 3</label><div class="book-tabs-content markdown-inner"><h5 id=distribution-of-cells-based-on-gene-level-counts>Distribution of cells based on gene level counts</h5><figure><img src=./testis3_vln.png width=80%></figure><h5 id=relationship-of-umi-vs-other-measures>Relationship of UMI vs other measures</h5><figure><img src=./testis3_scatter.png width=80%></figure><h5 id=grid-search-summary>Grid search summary</h5><figure><a href=./testis3_heatmap_grid.svg target=_blank><img src=./testis3_heatmap_grid.svg alt="click image to enlarge" width=100%></a><figcaption><p>click image to enlarge</p></figcaption></figure></div><input type=radio name="tabs-Gene Level Filters" style=display:none id="tabs-Gene Level Filters-3">
<label for="tabs-Gene Level Filters-3">Testis 4</label><div class="book-tabs-content markdown-inner"><h5 id=distribution-of-cells-based-on-gene-level-counts>Distribution of cells based on gene level counts</h5><figure><img src=./testis4_vln.png width=80%></figure><h5 id=relationship-of-umi-vs-other-measures>Relationship of UMI vs other measures</h5><figure><img src=./testis4_scatter.png width=80%></figure><h5 id=grid-search-summary>Grid search summary</h5><figure><a href=./testis4_heatmap_grid.svg target=_blank><img src=./testis4_heatmap_grid.svg alt="click image to enlarge" width=100%></a><figcaption><p>click image to enlarge</p></figcaption></figure></div></div><p></p><h3 id=conclusions>Conclusions</h3><p>In summary I applied the following approaches:</p><ul><li><strong>Removed Empty GEMs:</strong> Using a consensus approach with <code>cellranger3-wf</code> and <code>DropLetUtils</code></li><li><strong>Removed heterotypic multiplets:</strong> Using an <em>in silico</em> approach with <code>scrublet</code>.</li><li><strong>Removed homotypic multiplets and low complexity cells:</strong> Using a sane cutoff of (200≤ Num Expressed Genes per Cell &lt;5,000) that does not change clustering too compared to similar criteria.</li></ul><p>For the final number of selected cells see <strong>Table 1</strong>.</p></article><div class="book-footer justify-between"><div><a href=https://github.com/jfear/larval_gonad_docs/commit/de6fd743ec025cb9aa96eaf5e52115831aa3303d title="Last modified Jun 24, 2019 by Justin M. Fear" target=_blank rel=noopener><img src=/larval_gonad_docs/svg/calendar.svg alt=Changed> Jun 24, 2019</a></div><div><a href=https://github.com/jfear/larval_gonad_docs/edit/master/content/docs/scrnaseq_prep/cell_selection/_index.md target=_blank rel=noopener><img src=/larval_gonad_docs/svg/edit.svg alt=Edit> Edit this page</a></div></div></div><aside class="book-toc level-4 fixed"><nav id=TableOfContents><ul><li><ul><li><a href=#cell-selection>Cell Selection</a><ul><li><a href=#tl-dr>TL;DR</a></li><li><a href=#overview>Overview</a></li><li><a href=#results>Results</a><ul><li><a href=#empty-gem-identification>Empty GEM Identification</a></li><li><a href=#heterotypic-doublet-detection>Heterotypic Doublet Detection</a></li><li><a href=#homotypic-doublet-and-low-complexity-detection>Homotypic Doublet and Low Complexity Detection</a><ul><li><a href=#cutoff-criteria>Cutoff Criteria</a></li></ul></li></ul></li><li><a href=#conclusions>Conclusions</a></li></ul></li></ul></li></ul></nav></aside></main></body></html>