<!doctype html><html lang=en><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><title>Cell Selection
| Larval Testis X Inactivation</title><link rel=stylesheet href=/larval_gonad_docs/book.min.c4ee552cc972ef71d5af01635e3098aaecd8beebdbc742ba939e26a7d7a86b0e.css><link rel=icon href=/larval_gonad_docs/favicon.png type=image/x-icon><link rel=alternate type=application/rss+xml href=https://jfear.github.io/larval_gonad_docs/docs/scrnaseq_prep/cell_selection/index.xml title="Larval Testis X Inactivation"></head><body><input type=checkbox style=display:none id=menu-control><main class="flex container"><aside class="book-menu fixed"><nav><h2 class=book-brand><a href=https://jfear.github.io/larval_gonad_docs>Larval Testis X Inactivation</a></h2><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/>Data Prep and Quality Control</a><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cell_selection/ class=active>Cell Selection</a><ul></ul></li><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/clustering/>Cell Clustering</a><ul></ul></li></ul></li></ul></nav><script>(function(){var menu=document.querySelector('aside.book-menu nav')
addEventListener('beforeunload',function(event){localStorage.setItem('menu.scrollTop',menu.scrollTop)});menu.scrollTop=localStorage.getItem('menu.scrollTop')})()</script></aside><div class=book-page><header class="align-center justify-between book-header"><label for=menu-control><img src=/larval_gonad_docs/svg/menu.svg alt=Menu></label>
<strong>Cell Selection</strong></header><article class=markdown><h1 id=cell-selection>Cell Selection</h1><h2 id=tl-dr>TL;DR</h2><p><strong>Problem:</strong> Identify individual cells from scRNA-Seq data.</p><p>Cell selection is variable with different parameters and versions.
Cell ranger v2 with default parameters under performs returning ~500 cells per replicate.
Cell ranger v2 with the <code>--force-cells</code> option forces the returning of the expected number of cells for each replicate, but this is rather <em>ad hoc</em>.
Cell ranger v3 with default options returns the expected number of cells or more.
This seems less <em>ad hoc</em> than using the <code>--force-cells</code> option, but maybe over calling the number of cells especially in replicates 2 and 3.
Also the addition of low count cells seems to muddle cluster calling in downstream analysis.
While similar clusters are identified, irrespective of this step, clusters called from cell ranger v2 <code>--force-cells</code> have better separation and representation of known markers from the literature.</p><h2 id=overview>Overview</h2><figure><img src="https://cdn.technologynetworks.com/tn/images/thumbs/webp/640_360/10x-genomics-extends-their-application-portfolio-305346.webp?v=9720678" alt="Figure 1. The 10X Library Preparation consists of mixing individual cells with individual gel beads." width=80%><figcaption><p><strong>Figure 1. The 10X Library Preparation consists of mixing individual cells with individual gel beads.</strong></p></figcaption></figure><p>After sequencing, the first step in scRNA-Seq analysis is to identify individual cells.
On the 10X Genomics platform, libraries are prepared by combining individual cells with individual beads (<strong>Figure 1.</strong>).
Each bead contains a unique barcode which is used for demultiplexing cells.
The goal of cell selection is to find beads that have 1 and only 1 cell.
The majority of beads will be empty because cells are loaded using a double Poisson distribution.
However, empty cells may still have signal due to RNA in the media.
In theoretically rare cases a bead may be combined with 2 or more cells.
This could be caused by incomplete dissociation or loading of 2 separate cells in the same bead.
This is particularly problematic when the 2+ cells are all from the same cell-type.</p><p>We visualized our single-cell preps after filtration (35 μm).
We do see some evidence of multiple small cells stuck together.
Because we are using a filter we are biasing our chance of capturing large cell aggregates.
In particular, fused later staged primary spermatocytes are too large (~50 μm) to pass through the filter.
Any single cell-type multiplet will be from the smaller somatic cells, spermatogonia, or early stage primary spermatocytes.</p><p>To demultiplex and identify individual cells 10X Genomics provides their own software (<a href=https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/algorithms/overview><code>cell ranger</code></a>).
We tried <code>cell ranger</code> versions <code>v2</code> and <code>v3</code>.
According to the 10X Genomics website, <code>v3</code> has improved calling low RNA content cells.
Both versions begin by calculating a total unique molecular index (UMI) cutoff based on distributional assumptions.
This step is good at identifying cells with high RNA content.
It performs poorly when multiple cell-types are present with a high degree of RNA content heterogeneity.
Our data contains a wide range of cell-types from high RNA-content primary spermatocytes to essentially quiescent stem cell populations.
Version 3 of the software adds an additional step of comparing the remaining putative &ldquo;empty&rdquo; cells to each other.
Based on this comparison they can separate low RNA content cells from empty GEMS.</p><p>After cell selection <code>cell ranger</code> aligns reads using <code>STAR</code> and provides the gene count matrix for downstream analysis.
It also provides summary statistics shown below.
According to 10X Genomics&rsquo; website, they expect a capture rate of ~50% of loaded cells.
<strong>Table 1</strong> shows our expectations as well as the number of reads sequenced for each replicate.</p><p><strong>Table 1. Expectations for cell capture and read counts.</strong></p><table><thead><tr><th>Rep</th><th>Approx Num Cells Loaded</th><th>Expected Num Cells Captured</th><th>Num of Reads</th></tr></thead><tbody><tr><td>Testis 1</td><td>6,000</td><td>3,000</td><td>119,932,060</td></tr><tr><td>Testis 2</td><td>6,000</td><td>3,000</td><td>148,921,414</td></tr><tr><td>Testis 3</td><td>16,000</td><td>8,000</td><td>131,590,010</td></tr></tbody></table><h2 id=results>Results</h2><h3 id=cell-ranger>Cell Ranger</h3><p>Cell ranger summaries are displayed below (<strong>click on tabs to view</strong>).</p><h4 id=v2-defaults>V2 Defaults</h4><p>Cell ranger identified ~500 cells per replicate.
This is less than 10% of loaded cells being recovered.
Looking at the barcode rank plots we see that cells are called slightly past the first inflection point.
We can also see that ~50% of reads are found within a cell.
Cells that are captured have a high RNA content and large number of gene expressed (&gt;4k).
The extra depth in replicates 2 and 3 provided roughly double the median number of genes per cell, even though the number of reads per cell increased modestly.
This would lead to an over all decrease in sparsity of the cell counts matrix.</p><h4 id=v2-force>V2 Force</h4><p>Using the force option we can increase the number of cells per replicate to approach the theoretical 50% capture rate (<strong>Table 1</strong>).
However, the added cells have a lower UMI and gene content.
We still see an affect of increased depth with Replicates 2 and 3 having twice as many genes expressed per cell.
Again the number of reads per cell was not affected by sequencing depth.
Forcing also increased the fraction of reads found in a cell to ~60%.</p><h4 id=v3-defaults>V3 Defaults</h4><p>The new version of cell ranger does call more cells than version 2 (defaults and force).
By adding so many low UMI cells, the replicate differences in gene content have all but disappeared.
While the fraction of reads per cell is still ~60%.
While this method is less <em>ad hoc</em> than V2 force, results are not appreciably better.</p><div class=book-tabs><input type=radio name="tabs-summary stats" style=display:none id="tabs-summary stats-0" checked>
<label for="tabs-summary stats-0">Cell Ranger (v2.1.1; defaults)</label><div class="book-tabs-content markdown-inner"><h4 id=testis-replicate-1>Testis Replicate 1</h4><figure><img src=cellranger2_testis1_summary.png width=100%></figure><h4 id=testis-replicate-2>Testis Replicate 2</h4><figure><img src=cellranger2_testis2_summary.png width=100%></figure><h4 id=testis-replicate-3>Testis Replicate 3</h4><figure><img src=cellranger2_testis3_summary.png width=100%></figure><p><a href=https://github.com/jfear/larval_gonad/blob/5f8f4569ea253ab96d497055e7ae6ebb4c82a744/scrnaseq-wf/Snakefile#L96-L121 target=_blank class=book-btn>Code</a></div><input type=radio name="tabs-summary stats" style=display:none id="tabs-summary stats-1">
<label for="tabs-summary stats-1">Cell Ranger (v2.1.1; force)</label><div class="book-tabs-content markdown-inner"><h4 id=testis-replicate-1>Testis Replicate 1</h4><figure><img src=cellranger2_testis1_force_summary.png width=100%></figure><h4 id=testis-replicate-2>Testis Replicate 2</h4><figure><img src=cellranger2_testis2_force_summary.png width=100%></figure><h4 id=testis-replicate-3>Testis Replicate 3</h4><figure><img src=cellranger2_testis3_force_summary.png width=100%></figure><p><a href=https://github.com/jfear/larval_gonad/blob/5f8f4569ea253ab96d497055e7ae6ebb4c82a744/scrnaseq-wf/Snakefile#L123-L151 target=_blank class=book-btn>Code</a></div><input type=radio name="tabs-summary stats" style=display:none id="tabs-summary stats-2">
<label for="tabs-summary stats-2">Cell Ranger (v3.0.1; defaults)</label><div class="book-tabs-content markdown-inner"><h4 id=testis-replicate-1>Testis Replicate 1</h4><figure><img src=cellranger3_testis1_summary.png width=100%></figure><h4 id=testis-replicate-2>Testis Replicate 2</h4><figure><img src=cellranger3_testis2_summary.png width=100%></figure><h4 id=testis-replicate-3>Testis Replicate 3</h4><figure><img src=cellranger3_testis3_summary.png width=100%></figure><p><a class=book-btn>Code</a></div></div><p></p><h3 id=clustering>Clustering</h3><p>By looking at the cell ranger summary statistics we can begin to get a feel for the data.
However, it is hard to understand if the added cells are high quality and improve results.
The next step in the analysis is clustering cells and identifying cell types.
To get a better understanding of how cell selection affect clustering I ran one clustering method on each cell selection method.
For more details on clustering see <a href=/larval_gonad_docs/docs/scrnaseq_prep/clustering/>Cell Clustering</a>.</p><div class=book-tabs><input type=radio name=tabs-clustering style=display:none id=tabs-clustering-0 checked>
<label for=tabs-clustering-0>Cell Ranger (v2.1.1; defaults)</label><div class="book-tabs-content markdown-inner"><p><a class=book-btn>Code</a></div><input type=radio name=tabs-clustering style=display:none id=tabs-clustering-1>
<label for=tabs-clustering-1>Cell Ranger (v2.1.1; force)</label><div class="book-tabs-content markdown-inner"><p><a class=book-btn>Code</a></div><input type=radio name=tabs-clustering style=display:none id=tabs-clustering-2>
<label for=tabs-clustering-2>Cell Ranger (v3.0.1; defaults)</label><div class="book-tabs-content markdown-inner"><p><a class=book-btn>Code</a></div></div><p></p><h2 id=conclusion>Conclusion</h2></article><div class="book-footer justify-between"><div><a href=https://github.com/jfear/larval_gonad_docs/commit/8c644aa79f2fb54fb5aa77a3beffc2dd069a2962 title="Last modified May 31, 2019 by Justin M. Fear" target=_blank rel=noopener><img src=/larval_gonad_docs/svg/calendar.svg alt=Changed> May 31, 2019</a></div><div><a href=https://github.com/jfear/larval_gonad_docs/edit/master/content/docs/scrnaseq_prep/cell_selection/_index.md target=_blank rel=noopener><img src=/larval_gonad_docs/svg/edit.svg alt=Edit> Edit this page</a></div></div></div><aside class="book-toc level-4 fixed"><nav id=TableOfContents><ul><li><a href=#cell-selection>Cell Selection</a><ul><li><a href=#tl-dr>TL;DR</a></li><li><a href=#overview>Overview</a></li><li><a href=#results>Results</a><ul><li><a href=#cell-ranger>Cell Ranger</a><ul><li><a href=#v2-defaults>V2 Defaults</a></li><li><a href=#v2-force>V2 Force</a></li><li><a href=#v3-defaults>V3 Defaults</a></li></ul></li><li><a href=#clustering>Clustering</a></li></ul></li><li><a href=#conclusion>Conclusion</a></li></ul></li></ul></nav></aside></main></body></html>