<!doctype html><html lang=en><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><title>Neo X Gene Movement
| Larval Testis X Inactivation</title><link rel=stylesheet href=/larval_gonad_docs/book.min.e0adbac1e0fc6fdc3ef34548c686f58a7e1ba9fb1adf5c9591aa4b7df3653f22.css><link rel=icon href=/larval_gonad_docs/favicon.png type=image/x-icon><link rel=alternate type=application/rss+xml href=https://jfear.github.io/larval_gonad_docs/docs/gene_movement/index.xml title="Larval Testis X Inactivation"></head><body><input type=checkbox style=display:none id=menu-control><main class="flex container"><aside class="book-menu fixed"><nav><h2 class=book-brand><a href=https://jfear.github.io/larval_gonad_docs>Larval Testis X Inactivation</a></h2><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/>Data Prep and Quality Control</a><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cell_selection/>Cell Selection</a><ul></ul></li><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cluster_selection/>Selecting Clustering Algorithm</a><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cluster_selection/seurat2/testis1/>Seurat2 Testis 1 Clustering</a></li></ul></li></ul></li><li><a href=/larval_gonad_docs/docs/clustering/>Clustering</a><ul><li><a href=/larval_gonad_docs/docs/clustering/combined_n3/>Integrated Clustering (n = 3)</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis1/>Testis 1 Individual Clustering</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis2/>Testis 2 Individual Clustering</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis3/>Testis 3 Individual Clustering</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis4/>Testis 4 Individual Clustering</a></li></ul></li><li><a href=/larval_gonad_docs/docs/intronless_genes/>Intronless Gene Expression</a><ul></ul></li><li><a href=/larval_gonad_docs/docs/gene_movement/ class=active>Neo X Gene Movement</a><ul></ul></li></ul></nav><script>(function(){var menu=document.querySelector('aside.book-menu nav')
addEventListener('beforeunload',function(event){localStorage.setItem('menu.scrollTop',menu.scrollTop)});menu.scrollTop=localStorage.getItem('menu.scrollTop')})()</script></aside><div class=book-page><header class="align-center justify-between book-header"><label for=menu-control><img src=/larval_gonad_docs/svg/menu.svg alt=Menu></label>
<strong>Neo X Gene Movement</strong></header><article class=markdown><p><span style="font-size:3em;font-weight:'strong';color:red">WARNING: Preliminary Analysis</span></p><h2 id=neo-x-gene-movement>Neo X Gene Movement</h2><p>We are interested in the evolutionary impact of X inactivation.
There are several hypotheses about why the X would evolve MSCI during spermatogenesis.
Here we focus on the idea that genes important for spermatogenesis will move off of the X to avoid silencing.
To investigate this question, we look at gene movement off of the Neo X chromosome in <em>D. pseudoobscura</em>.
In <em>D. pseudoobscura</em> Muller element A (X) fused with Muller element D (3L) (<strong>Figure 1</strong>).
We expect that genes important for spermatogenesis, on Muller element D, will move to another chromosome or be lost.</p><figure><img src=drosophila_synteny.png alt="Figure 1: Chromosome synteny across the Drosophila genus. Image modified from http://flybase.org/maps/synteny." width=100%><figcaption><p><b>Figure 1: Chromosome synteny across the Drosophila genus.</b> Image modified from <a href=http://flybase.org/maps/synteny>http://flybase.org/maps/synteny</a>.</p></figcaption></figure><p>First, we need to characterize gene movement in the Drosophila genus.
Then we can look for enrichment in gene movement off of Muller D in genes up regulated in primary spermatocytes.
To characterize gene movement we look at gene localization in a subset of the Drosophila genus.
Here we focus on 5 members of the genus: <em>D. pseudoobscura</em>, <em>D. melanogaster</em>, <em>D. willistoni</em>, <em>D. mojavensis</em>, and <em>D. virilis</em>.
We downloaded updated gene annotations and ortholog mappings from <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE99574">GSE99574</a>.
We selected a set of 12,365 orthologous genes present in <em>D. melanogaster</em> and at least one of the other species.
Using these gene locations we can determine how the gene is behaving in <em>D. pseudoobscura</em>.</p><h3 id=methods>Methods</h3><h4 id=identification-of-orthologs>Identification of orthologs</h4><p>The quality of assemblies and annotations are highly variable across the Drosophila genus.
We found that genes from <em>D. willistoni</em>, <em>D. mojavensis</em>, and <em>D. virilis</em> were mapped to scaffolds instead of chromosome arm.
We used annotations from <em>D. melanogaster</em> to assign Muller element to each of these scaffolds using a consensus approach.
For each scaffold, with &gt; 20 genes, we looked where their orthologs were located in <em>D. melanogaster</em>.
We then assigned the Muller element using majority rule.
To validate our approach, we did this same process for <em>D. pseudoobscura</em> which has chromosomal annotation.
We achieved 100% agreement with <em>D. pseudoobscura</em> annotation.</p><h4 id=classification-of-evolutionary-state>Classification of evolutionary state</h4><p>We classified each gene as conserved, gene_death, moved_on, or moved_off based on localization across the genus (<strong>Table 1</strong>).
For example, if a gene is located on Muller D in all species then it is conserved (<strong>Table 1a</strong>).
Similarly, recent_conserved genes were on Muller D in <em>Sophophora</em> branch, but else where in <em>D. virilis</em> and <em>D. mojavensis</em>.
If the gene was on Muller D in all species, but not found in <em>D. pseudoobscura</em> then it was considered gene_death.
If the gene was on Muller D in all species, but on another Muller element in <em>D. pseudoobscura</em> then it was considered moved_off.
We then furthered grouped genes as selection favoring a gene remaining where it is (conserved, moved_on) or selection against where it is (gene_death, moved_off).</p><p><strong>Table 1a. Classification of genes on Muller element D.</strong></p><table><thead><tr><th>Dmel</th><th>Dpse</th><th>(Dwil/Dvir/Dmoj)</th><th>Class</th><th>Group</th></tr></thead><tbody><tr><td>D</td><td>D</td><td>D</td><td>conserved</td><td>Selection Favor D</td></tr><tr><td>Other</td><td>D</td><td>Other</td><td>moved_on</td><td>Selection Favor D</td></tr><tr><td>D</td><td>Other</td><td>D</td><td>moved_off</td><td>Selection Against D</td></tr><tr><td>D</td><td>Not_found</td><td>D</td><td>gene_death</td><td>Selection Against D</td></tr></tbody></table><p><strong>Table 1b. Classification of genes on Muller element E.</strong></p><table><thead><tr><th>Dmel</th><th>Dpse</th><th>(Dwil/Dvir/Dmoj)</th><th>Class</th><th>Group</th></tr></thead><tbody><tr><td>E</td><td>E</td><td>E</td><td>conserved</td><td>Selection Favor E</td></tr><tr><td>Other</td><td>E</td><td>Other</td><td>moved_on</td><td>Selection Favor E</td></tr><tr><td>E</td><td>Other</td><td>E</td><td>moved_off</td><td>Selection Against E</td></tr><tr><td>E</td><td>Not_found</td><td>E</td><td>gene_death</td><td>Selection Against E</td></tr></tbody></table><h3 id=results>Results</h3><p>We focused on genes located on Muller elements A (X), D (3L), and E (3R).
The majority of genes were conserved with very few genes moving onto the respective Muller element (<strong>Table 2</strong>).
Gene death in <em>D. pseudoobscura</em> maybe overestimated due to quality of gene annotation.</p><p><strong>Table 2. Number of genes per Muller Element.</strong></p><table><thead><tr><th></th><th>Muller A</th><th>Muller D</th><th>Muller E</th></tr></thead><tbody><tr><td><b>conserved</b></td><td>1,079</td><td>1,235</td><td>1,735</td></tr><tr><td><b>recent_conserved</b></td><td>246</td><td>337</td><td>291</td></tr><tr><td><b>moved_on</b></td><td>7</td><td>4</td><td>4</td></tr><tr><td><b>gene_death</b></td><td>375</td><td>575</td><td>647</td></tr><tr><td><b>moved_off</b></td><td>135</td><td>158</td><td>193</td></tr><tr><td><b>other</b></td><td>284</td><td>386</td><td>468</td></tr></tbody></table><h4 id=distribution-of-primary-spermatocyte-expression>Distribution of primary spermatocyte expression</h4><p>We looked at the proportion of primary spermatocyte cells that expressed orthologous genes in each evolutionary class (<strong>Figure 1,2,3</strong>).
The number of genes in the moved_on and moved_off classes are typically very low (&lt; 10) which makes interpretation difficult.
If we group genes into &ldquo;Selection Favor&rdquo; and &ldquo;Selection Against&rdquo;, followed by a Mann-Whitney U test, we see an significant increased spermatocyte expression for genes with &ldquo;Selection Against&rdquo; for all three Muller elements.
These differences appear to be driven by conserved genes and genes that undergone gene death.
Interestingly, the percent of spermatocytes expressing conserved genes is lower for Muller A (X) than D or E.</p><div class=book-tabs><input type=radio name=tabs-cyte_dist style=display:none id=tabs-cyte_dist-0 checked>
<label for=tabs-cyte_dist-0>Muller A</label><div class="book-tabs-content markdown-inner"><figure><img src=./2019-07-22_neox_analysis_boxplot_mullerA.svg alt="Figure 2. Gene movement on/off of Muller element A." width=100%><figcaption><p><b>Figure 2. Gene movement on/off of Muller element A.</b></p></figcaption></figure><p>Selection Favor vs Selection Against: <em>Mann-Whitney-U p-value: 1.32e-05</em></p></div><input type=radio name=tabs-cyte_dist style=display:none id=tabs-cyte_dist-1>
<label for=tabs-cyte_dist-1>Muller D</label><div class="book-tabs-content markdown-inner"><figure><img src=./2019-07-22_neox_analysis_boxplot_mullerD.svg alt="Figure 3. Gene movement on/off of Muller element D." width=100%><figcaption><p><b>Figure 3. Gene movement on/off of Muller element D.</b></p></figcaption></figure><p>Selection Favor vs Selection Against: <em>Mann-Whitney-U p-value: 4.78e-12</em></p></div><input type=radio name=tabs-cyte_dist style=display:none id=tabs-cyte_dist-2>
<label for=tabs-cyte_dist-2>Muller E</label><div class="book-tabs-content markdown-inner"><figure><img src=./2019-07-22_neox_analysis_boxplot_mullerE.svg alt="Figure 4. Gene movement on/off of Muller element D." width=100%><figcaption><p><b>Figure 4. Gene movement on/off of Muller element D.</b></p></figcaption></figure><p>Selection Favor vs Selection Against: <em>Mann-Whitney-U p-value: 3.75e-17</em></p></div></div><p></p><h4 id=enrichment-of-primary-spermatocyte-biased-genes>Enrichment of primary spermatocyte biased genes</h4><p>To how genes important for spermatogenesis behave we looked at differential expression between Spermatogonia and Primary Spermatoctyes.
Below are cross tabulated tables showing counts for the different categories: spermatogonia biased gene expression (gonia), primary spermatocyte gene expression (cyte), and not differentially expressed genes (NS).
Again counts are very low for some classes, but I performed a chi-squre test followed by posthoc analysis to identify enrichment or depletion.
To improve cell counts, I aggregated samples to &ldquo;Cyte Biased&rdquo; and &ldquo;Not Biased&rdquo; and grouped evolutionary classes.
Again there is evidence of a enrichment of genes that underwent gene_death or &ldquo;Selection Against&rdquo; for all tested Muller elements.
While all Muller elements were significant, Muller D shows a more extreme pattern.</p><div class=book-tabs><input type=radio name=tabs-crosstabs style=display:none id=tabs-crosstabs-0 checked>
<label for=tabs-crosstabs-0>Muller A</label><div class="book-tabs-content markdown-inner"><p><strong>Table 3. Number of genes with differential expression on Muller A.</strong></p><table><thead><tr><th>DEG</th><th>conserved</th><th>moved_on</th><th>gene_death</th><th>moved_off</th></tr></thead><tbody><tr><td>NS</td><td>0<sup>*</sup></td><td>1<sup>†</sup></td><td>4</td><td>1</td></tr><tr><td>cyte</td><td>1<sup>*</sup></td><td>1</td><td>9<sup>†</sup></td><td>1</td></tr><tr><td>gonia</td><td>62<sup>†</sup></td><td>0<sup>*</sup></td><td>19<sup>*</sup></td><td>4</td></tr></tbody></table><p><span style=font-size:.8em>𝛘^2: 35.4513, p-value: 0.0000, df: 6</span><br><span style=font-size:.8em><sup>*</sup>Significantly depleted (p-value ≤ 0.01)</span><br><span style=font-size:.8em><sup>†</sup>Significantly enriched (p-value ≤ 0.01)</span></p><p><strong>Table 4. Number of genes with selection on Muller A.</strong></p><table><thead><tr><th>DEG</th><th>Selection Favor</th><th>Selection Against</th></tr></thead><tbody><tr><td>Cyte Biased</td><td>2</td><td>10</td></tr><tr><td>Not Biased</td><td>63</td><td>28</td></tr></tbody></table><p><span style=font-size:.8em>Fisher&rsquo;s exact test p-value: 0.0007</span></p></div><input type=radio name=tabs-crosstabs style=display:none id=tabs-crosstabs-1>
<label for=tabs-crosstabs-1>Muller D</label><div class="book-tabs-content markdown-inner"><p><strong>Table 5. Number of genes with differential expression on Muller D.</strong></p><table><thead><tr><th>DEG</th><th>conserved</th><th>gene_death</th><th>moved_off</th></tr></thead><tbody><tr><td>NS</td><td>8</td><td>7</td><td>1</td></tr><tr><td>cyte</td><td>17<sup>*</sup></td><td>27<sup>†</sup></td><td>7</td></tr><tr><td>gonia</td><td>86<sup>†</sup></td><td>7<sup>*</sup></td><td>4</td></tr></tbody></table><p><span style=font-size:.8em>𝛘^2: 50.6224, p-value: 0.0000, df: 4</span><br><span style=font-size:.8em><sup>*</sup>Significantly depleted (p-value ≤ 0.01)</span><br><span style=font-size:.8em><sup>†</sup>Significantly enriched (p-value ≤ 0.01)</p><p><strong>Table 6. Number of genes with selection on Muller D.</strong></p><table><thead><tr><th>DEG</th><th>Selection Favor</th><th>Selection Against</th></tr></thead><tbody><tr><td>Cyte Biased</td><td>17</td><td>34</td></tr><tr><td>Not Biased</td><td>94</td><td>19</td></tr></tbody></table><p><span style=font-size:.8em>Fisher&rsquo;s exact test p-value: 6.461e-10</span></p></div><input type=radio name=tabs-crosstabs style=display:none id=tabs-crosstabs-2>
<label for=tabs-crosstabs-2>Muller E</label><div class="book-tabs-content markdown-inner"><p><strong>Table 7. Number of genes with differential expression on Muller E.</strong></p><table><thead><tr><th>DEG</th><th>conserved</th><th>moved_on</th><th>gene_death</th><th>moved_off</th></tr></thead><tbody><tr><td>NS</td><td>13</td><td>0</td><td>5</td><td>2</td></tr><tr><td>cyte</td><td>28<sup>*</sup></td><td>1</td><td>28<sup>†</sup></td><td>10</td></tr><tr><td>gonia</td><td>117<sup>†</sup></td><td>0</td><td>12<sup>*</sup></td><td>11</td></tr></tbody></table><p><span style=font-size:.8em>𝛘^2: 41.5756, p-value: 0.0000, df: 6</span><br><span style=font-size:.8em><sup>*</sup>Significantly depleted (p-value ≤ 0.01)</span><br><span style=font-size:.8em><sup>†</sup>Significantly enriched (p-value ≤ 0.01)</span><br></p><p><strong>Table 8. Number of genes with selection on Muller E.</strong></p><table><thead><tr><th>DEG</th><th>Selection Favor</th><th>Selection Against</th></tr></thead><tbody><tr><td>Cyte Biased</td><td>29</td><td>38</td></tr><tr><td>Not Biased</td><td>130</td><td>30</td></tr></tbody></table><p><span style=font-size:.8em>Fisher&rsquo;s exact test p-value: 3.264e-08</span></p></div></div><p></p></article><div class="book-footer justify-between"><div><a href=https://github.com/jfear/larval_gonad_docs/commit/31c94d7028ac3119f62734bdc4266f6790a2c176 title="Last modified Jul 23, 2019 by Justin Fear" target=_blank rel=noopener><img src=/larval_gonad_docs/svg/calendar.svg alt=Changed> Jul 23, 2019</a></div><div><a href=https://github.com/jfear/larval_gonad_docs/edit/master/content/docs/gene_movement/_index.md target=_blank rel=noopener><img src=/larval_gonad_docs/svg/edit.svg alt=Edit> Edit this page</a></div></div></div><aside class="book-toc level-4 fixed"><nav id=TableOfContents><ul><li><ul><li><a href=#neo-x-gene-movement>Neo X Gene Movement</a><ul><li><a href=#methods>Methods</a><ul><li><a href=#identification-of-orthologs>Identification of orthologs</a></li><li><a href=#classification-of-evolutionary-state>Classification of evolutionary state</a></li></ul></li><li><a href=#results>Results</a><ul><li><a href=#distribution-of-primary-spermatocyte-expression>Distribution of primary spermatocyte expression</a></li><li><a href=#enrichment-of-primary-spermatocyte-biased-genes>Enrichment of primary spermatocyte biased genes</a></li></ul></li></ul></li></ul></li></ul></nav></aside></main></body></html>