<!doctype html><html lang=en><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><title>Neo X Gene Movement
| Larval Testis X Inactivation</title><link rel=stylesheet href=/larval_gonad_docs/book.min.e0adbac1e0fc6fdc3ef34548c686f58a7e1ba9fb1adf5c9591aa4b7df3653f22.css><link rel=icon href=/larval_gonad_docs/favicon.png type=image/x-icon><link rel=alternate type=application/rss+xml href=https://jfear.github.io/larval_gonad_docs/docs/gene_movement/index.xml title="Larval Testis X Inactivation"></head><body><input type=checkbox style=display:none id=menu-control><main class="flex container"><aside class="book-menu fixed"><nav><h2 class=book-brand><a href=https://jfear.github.io/larval_gonad_docs>Larval Testis X Inactivation</a></h2><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/>Data Prep and Quality Control</a><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cell_selection/>Cell Selection</a><ul></ul></li><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cluster_selection/>Selecting Clustering Algorithm</a><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cluster_selection/seurat2/testis1/>Seurat2 Testis 1 Clustering</a></li></ul></li></ul></li><li><a href=/larval_gonad_docs/docs/clustering/>Clustering</a><ul><li><a href=/larval_gonad_docs/docs/clustering/combined_n3/>Integrated Clustering (n = 3)</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis1/>Testis 1 Individual Clustering</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis2/>Testis 2 Individual Clustering</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis3/>Testis 3 Individual Clustering</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis4/>Testis 4 Individual Clustering</a></li></ul></li><li><a href=/larval_gonad_docs/docs/intronless_genes/>Intronless Gene Expression</a><ul></ul></li><li><a href=/larval_gonad_docs/docs/gene_movement/ class=active>Neo X Gene Movement</a><ul><li><a href=/larval_gonad_docs/docs/gene_movement/dpse_gonia_vs_cytes/>D. Pseudoobscura (SP vs Cytes)</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dpse_gonia_vs_eps/>D. Pseudoobscura (SP vs EPS)</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dpse_gonia_vs_ps/>D. Pseudoobscura (SP vs PS)</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dpse_eps_vs_ps/>D. Pseudoobscura (EPS vs PS)</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dwil_gonia_vs_cytes/>D. willistoni (SP vs Cytes)</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dwil_gonia_vs_eps/>D. willistoni (SP vs EPS)</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dwil_gonia_vs_ps/>D. willistoni (SP vs PS)</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dwil_eps_vs_ps/>D. willistoni (EPS vs PS)</a></li></ul></li></ul></nav><script>(function(){var menu=document.querySelector('aside.book-menu nav')
addEventListener('beforeunload',function(event){localStorage.setItem('menu.scrollTop',menu.scrollTop)});menu.scrollTop=localStorage.getItem('menu.scrollTop')})()</script></aside><div class=book-page><header class="align-center justify-between book-header"><label for=menu-control><img src=/larval_gonad_docs/svg/menu.svg alt=Menu></label>
<strong>Neo X Gene Movement</strong></header><article class=markdown><p><span style="font-size:3em;font-weight:'strong';color:red">WARNING: Preliminary Analysis</span></p><h2 id=neo-x-gene-movement>Neo X Gene Movement</h2><p>We are interested in the evolutionary impact of X inactivation.
There are several hypotheses about why the X would evolve MSCI during spermatogenesis.
Here we focus on the idea that genes important for spermatogenesis will move off of the X to avoid silencing.
To investigate this question we look at gene movement off of the Neo X chromosome in <em>D. pseudoobscura</em> and <em>D. willistoni</em>.
In this group, the Muller element A (X) fused with Muller element D (3L) (<strong>Figure 1</strong>).
We expect that genes important for spermatogenesis, on Muller element D, will move to another chromosome or be lost.</p><figure><img src=drosophila_synteny.png alt="Figure 1: Chromosome synteny across the Drosophila genus. Image modified from http://flybase.org/maps/synteny." width=100%><figcaption><p><b>Figure 1: Chromosome synteny across the Drosophila genus.</b> Image modified from <a href=http://flybase.org/maps/synteny>http://flybase.org/maps/synteny</a>.</p></figcaption></figure><p>First, we need to characterize gene movement in the Drosophila genus.
Then we can look for enrichment of gene movement off of Muller D in genes up regulated in primary spermatocytes.
To characterize gene movement we look at gene localization in a subset of the Drosophila genus.
We focused on 5 members of the genus: <em>D. pseudoobscura</em>, <em>D. melanogaster</em>, <em>D. willistoni</em>, <em>D. mojavensis</em>, and <em>D. virilis</em>.
I downloaded updated gene annotations and ortholog mappings from <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE99574">GSE99574</a>.
I supplemented these ortholog mappings with those provided by
<a href=ftp://ftp.flybase.net/releases/FB2018_05/precomputed_files/orthologs/dmel_orthologs_in_drosophila_species_fb_2018_05.tsv.gz>FlyBase</a>.
We selected a set of 13,473 orthologous genes present in <em>D. melanogaster</em> and at least one of the other species.
Using these gene locations we can determine how the gene is behaving in <em>D. pseudoobscura</em> and <em>D. willistoni</em>.</p><h3 id=methods>Methods</h3><h4 id=identification-of-orthologs>Identification of orthologs</h4><p>The quality of assemblies and annotations are highly variable across the Drosophila genus.
Genes from <em>D. willistoni</em>, <em>D. mojavensis</em>, and <em>D. virilis</em> are currently mapped to scaffolds instead of chromosome arm.
We used annotations from <em>D. melanogaster</em> to assign Muller element to each of these scaffolds using a consensus approach.
For each scaffold, with &gt; 20 genes, we looked where the orthologs were located in <em>D. melanogaster</em>.
We assigned the Muller element using majority rule.
To validate our approach, we did this same process for <em>D. pseudoobscura</em> which has chromosomal annotation.
We achieved 100% agreement with <em>D. pseudoobscura</em> annotation.</p><h4 id=classification-of-evolutionary-state>Classification of evolutionary state</h4><p>We classified each gene as conserved, gene_death, moved_on, or moved_off based on localization across the genus (<strong>Table 1</strong>).
For example, if a gene is located on Muller D in all species then it is conserved (<strong>Table 1a</strong>).
Similarly, recent_conserved genes were on Muller D in <em>Sophophora</em> branch, but were elsewhere in <em>D. virilis</em> and <em>D. mojavensis</em>.
To determine movement in <em>D pseudoobscura</em> we did the following.
If the gene was on Muller D in all species, but not found in <em>D. pseudoobscura</em> then it was considered gene_death.
If the gene was on Muller D in all species, but on another Muller element in <em>D. pseudoobscura</em> then it was considered moved_off.
We used the same logic for <em>D. willistoni</em>.
We then furthered grouped genes as selection favoring a gene remaining where it is (conserved, moved_on) or selection against where it is (gene_death, moved_off).</p><p><strong>Table 1a. Classification of genes on Muller element D.</strong></p><table><thead><tr><th>Dmel</th><th>Dpse</th><th>(Dwil/Dvir/Dmoj)</th><th>Class</th><th>Group</th></tr></thead><tbody><tr><td>D</td><td>D</td><td>D</td><td>conserved</td><td>Selection Favor D</td></tr><tr><td>Other</td><td>D</td><td>Other</td><td>moved_on</td><td>Selection Favor D</td></tr><tr><td>D</td><td>Other</td><td>D</td><td>moved_off</td><td>Selection Against D</td></tr><tr><td>D</td><td>Not_found</td><td>D</td><td>gene_death</td><td>Selection Against D</td></tr></tbody></table><p><strong>Table 1b. Classification of genes on Muller element E.</strong></p><table><thead><tr><th>Dmel</th><th>Dpse</th><th>(Dwil/Dvir/Dmoj)</th><th>Class</th><th>Group</th></tr></thead><tbody><tr><td>E</td><td>E</td><td>E</td><td>conserved</td><td>Selection Favor E</td></tr><tr><td>Other</td><td>E</td><td>Other</td><td>moved_on</td><td>Selection Favor E</td></tr><tr><td>E</td><td>Other</td><td>E</td><td>moved_off</td><td>Selection Against E</td></tr><tr><td>E</td><td>Not_found</td><td>E</td><td>gene_death</td><td>Selection Against E</td></tr></tbody></table><h3 id=results>Results</h3><h4 id=ortholog-selection>Ortholog selection</h4><p>Yang et al. only generated orthologs based on genes expressed in their data.
There were a number of FlyBase annotated orthologs that were excluded from their list.
The majority of orthologs present in both data sets were mapping to the same Muller element (<strong>Table 2a</strong>).
While the similarity of missing orthologs was very different (<strong>Table 2a</strong>).
I merged these ortholog sets by updating the FlyBase annotation using Yang et al. mappings (<strong>Table 2a</strong>).
The number of genes present on each Muller element is similar across species with most elements having ≤ 175 different (<strong>Table 2b</strong>).</p><p><strong>Table 2a. Similarity of Yang et al. vs FlyBase.</strong></p><table><thead><tr><th>species</th><th>Num Non Missing</th><th>Similarity of Non Missing<sup>†</sup></th><th>Num Yang Missing</th><th>Num FlyBase Missing</th><th>Similarity of Missing<sup>†</sup></th><th>Number Orthologs</th></tr></thead><tbody><tr><td>dmel</td><td>12,599</td><td>1</td><td>961</td><td>887</td><td>-0.0573347</td><td>13,559</td></tr><tr><td>dpse</td><td>9,533</td><td>0.987294</td><td>4,597</td><td>2,250</td><td>0.301446</td><td>12,358</td></tr><tr><td>dvir</td><td>9,657</td><td>0.993625</td><td>4,427</td><td>2,663</td><td>0.393496</td><td>11,948</td></tr><tr><td>dmoj</td><td>9,747</td><td>0.994893</td><td>4,316</td><td>2,772</td><td>0.427417</td><td>11,855</td></tr><tr><td>dwil</td><td>9,751</td><td>0.987739</td><td>4,337</td><td>2,544</td><td>0.384694</td><td>12,102</td></tr></tbody></table><p><sup>†</sup><a href=https://scikit-learn.org/stable/modules/generated/sklearn.metrics.adjusted_rand_score.html>Adj. Rand Score</a>.</p><p><strong>Table 2b. Number of genes per muller element.</strong></p><table><thead><tr><th>Species</th><th>A</th><th>B</th><th>C</th><th>D</th><th>E</th><th>F</th><th>Unknown Scaffold</th></tr></thead><tbody><tr><td>dmel</td><td>2,102</td><td>2,582</td><td>2,766</td><td>2,656</td><td>3,351</td><td>78</td><td>24</td></tr><tr><td>dpse</td><td>1,850</td><td>2,281</td><td>2,458</td><td>2,310</td><td>3,087</td><td>0</td><td>372</td></tr><tr><td>dvir</td><td>1,913</td><td>2,184</td><td>2,419</td><td>2,316</td><td>2,930</td><td>77</td><td>109</td></tr><tr><td>dmoj</td><td>1,858</td><td>2,222</td><td>2,398</td><td>2,262</td><td>2,970</td><td>93</td><td>52</td></tr><tr><td>dwil</td><td>1,984</td><td>2,175</td><td>2,489</td><td>2,219</td><td>3,075</td><td>0</td><td>160</td></tr><tr><td><strong>Std Dev</strong></td><td>104.55</td><td>169.13</td><td>149.52</td><td>174.11</td><td>164.38</td><td>45.72</td><td>138.14</td></tr></tbody></table><h4 id=gene-movement-classification>Gene movement classification</h4><p>For gene movement results browse to the respective page either using the links below or the side bar.</p><p><strong>Table 3. <em>D. pseudoobscura</em> movement classes.</strong></p><table><thead><tr><th></th><th>muller_A</th><th>muller_D</th><th>muller_E</th></tr></thead><tbody><tr><td>conserved</td><td>1,571</td><td>1,919</td><td>2,557</td></tr><tr><td>recent_conserved</td><td>140</td><td>159</td><td>214</td></tr><tr><td>moved_on</td><td>21</td><td>10</td><td>19</td></tr><tr><td>gene_death</td><td>245</td><td>373</td><td>390</td></tr><tr><td>moved_off</td><td>64</td><td>55</td><td>44</td></tr><tr><td>other</td><td>247</td><td>351</td><td>433</td></tr></tbody></table><p><strong>Table 4. <em>D. willistoni</em> movement classes.</strong></p><table><thead><tr><th></th><th>muller_A</th><th>muller_D</th><th>muller_E</th></tr></thead><tbody><tr><td>conserved</td><td>1,571</td><td>1,919</td><td>2,557</td></tr><tr><td>recent_conserved</td><td>140</td><td>159</td><td>214</td></tr><tr><td>moved_on</td><td>0</td><td>0</td><td>0</td></tr><tr><td>gene_death</td><td>256</td><td>388</td><td>487</td></tr><tr><td>moved_off</td><td>32</td><td>141</td><td>81</td></tr><tr><td>other</td><td>289</td><td>260</td><td>318</td></tr></tbody></table><h4 id=gene-movement-cell-type-enrichment>Gene movement cell type enrichment</h4><h5 id=d-pseudoobscura>D. pseudoobscura</h5><ul><li><a href=/larval_gonad_docs/docs/gene_movement/dpse_gonia_vs_cytes/>SP vs Primary Spermatocytes</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dpse_gonia_vs_eps/>SP vs EPS</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dpse_gonia_vs_ps/>SP vs PS</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dpse_eps_vs_ps/>EPS vs PS</a></li></ul><h5 id=d-willistoni>D. willistoni</h5><ul><li><a href=/larval_gonad_docs/docs/gene_movement/dwil_gonia_vs_cytes/>SP vs Primary Spermatocytes</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dwil_gonia_vs_eps/>SP vs EPS</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dwil_gonia_vs_ps/>SP vs PS</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dwil_eps_vs_ps/>EPS vs PS</a></li></ul></article><div class="book-footer justify-between"><div><a href=https://github.com/jfear/larval_gonad_docs/commit/66fbe104adc4fc9530466c0160222c64f8817cf7 title="Last modified Aug 9, 2019 by Justin M. Fear" target=_blank rel=noopener><img src=/larval_gonad_docs/svg/calendar.svg alt=Changed> Aug 9, 2019</a></div><div><a href=https://github.com/jfear/larval_gonad_docs/edit/master/content/docs/gene_movement/_index.md target=_blank rel=noopener><img src=/larval_gonad_docs/svg/edit.svg alt=Edit> Edit this page</a></div></div></div><aside class="book-toc level-4 fixed"><nav id=TableOfContents><ul><li><ul><li><a href=#neo-x-gene-movement>Neo X Gene Movement</a><ul><li><a href=#methods>Methods</a><ul><li><a href=#identification-of-orthologs>Identification of orthologs</a></li><li><a href=#classification-of-evolutionary-state>Classification of evolutionary state</a></li></ul></li><li><a href=#results>Results</a><ul><li><a href=#ortholog-selection>Ortholog selection</a></li><li><a href=#gene-movement-classification>Gene movement classification</a></li><li><a href=#gene-movement-cell-type-enrichment>Gene movement cell type enrichment</a><ul><li><a href=#d-pseudoobscura>D. pseudoobscura</a></li><li><a href=#d-willistoni>D. willistoni</a></li></ul></li></ul></li></ul></li></ul></li></ul></nav></aside></main></body></html>