<!doctype html><html lang=en><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><title>Neo X Gene Movement
| Larval Testis X Inactivation</title><link rel=stylesheet href=/larval_gonad_docs/book.min.e0adbac1e0fc6fdc3ef34548c686f58a7e1ba9fb1adf5c9591aa4b7df3653f22.css><link rel=icon href=/larval_gonad_docs/favicon.png type=image/x-icon><link rel=alternate type=application/rss+xml href=https://jfear.github.io/larval_gonad_docs/docs/gene_movement/index.xml title="Larval Testis X Inactivation"></head><body><input type=checkbox style=display:none id=menu-control><main class="flex container"><aside class="book-menu fixed"><nav><h2 class=book-brand><a href=https://jfear.github.io/larval_gonad_docs>Larval Testis X Inactivation</a></h2><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/>Data Prep and Quality Control</a><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cell_selection/>Cell Selection</a><ul></ul></li><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cluster_selection/>Selecting Clustering Algorithm</a><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cluster_selection/seurat2/testis1/>Seurat2 Testis 1 Clustering</a></li></ul></li></ul></li><li><a href=/larval_gonad_docs/docs/clustering/>Clustering</a><ul><li><a href=/larval_gonad_docs/docs/clustering/combined_n3/>Integrated Clustering (n = 3)</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis1/>Testis 1 Individual Clustering</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis2/>Testis 2 Individual Clustering</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis3/>Testis 3 Individual Clustering</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis4/>Testis 4 Individual Clustering</a></li></ul></li><li><a href=/larval_gonad_docs/docs/gene_movement/ class=active>Neo X Gene Movement</a><ul></ul></li></ul></nav><script>(function(){var menu=document.querySelector('aside.book-menu nav')
addEventListener('beforeunload',function(event){localStorage.setItem('menu.scrollTop',menu.scrollTop)});menu.scrollTop=localStorage.getItem('menu.scrollTop')})()</script></aside><div class=book-page><header class="align-center justify-between book-header"><label for=menu-control><img src=/larval_gonad_docs/svg/menu.svg alt=Menu></label>
<strong>Neo X Gene Movement</strong></header><article class=markdown><p><span style="font-size:3em;font-weight:'strong';color:red">WARNING: Preliminary Analysis</span></p><h2 id=neo-x-gene-movement>Neo X Gene Movement</h2><p>We are interested in the evolutionary impact of X inactivation.
There are several hypotheses about why the X would evolve MSCI during spermatogenesis.
Here we are focusing on the idea that if the X is really silenced during spermatogenesis, then genes important for male fertility should move off of the X.
During Drosophila genus evolution there was a fusion of Muller element A (X) with Muller element D (3L) in <em>D. pseudoobscura</em> (<strong>Figure 1</strong>).
Here we look at gene movement on <em>D. pseudoobscura</em> Muller element D.</p><figure><img src=drosophila_synteny.png alt="Figure 1: Chromosome synteny across the Drosophila genus. Image modified from http://flybase.org/maps/synteny." width=100%><figcaption><p><b>Figure 1: Chromosome synteny across the Drosophila genus.</b> Image modified from <a href=http://flybase.org/maps/synteny>http://flybase.org/maps/synteny</a>.</p></figcaption></figure><h3 id=methods>Methods</h3><h4 id=identification-of-orthologs>Identification of orthologs</h4><p>First, we looked at gene localization in <em>D. pseudoobscura</em>, <em>D. melanogaster</em>, <em>D. willistoni</em>, <em>D. mojavensis</em>, and <em>D. virilis</em>.
We downloaded updated gene annotations and ortholog tables from <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE99574">GSE99574</a>.
We selected a set of 12,365 orthologous genes that are present in <em>D. melanogaster</em> and at least one of the other species.
In this annotation, genes from <em>D. willistoni</em>, <em>D. mojavensis</em>, and <em>D. virilis</em> are mapped to scaffolds instead of chromosome arm.
To map genes to Muller element we used a consensus approach.
For each scaffold, with &gt; 20 genes, we lifted over the <em>D. melanogaster</em> Muller annotation using majority rule.
To validate our approach, we did this same process for <em>D. pseudoobscura</em> which has chromosomal annotation.
We achieved 100% agreement with <em>D. pseudoobscura</em> annotation.</p><h4 id=differential-expression>Differential expression</h4><h3 id=results>Results</h3><p><strong>Table 1. Number of genes per Muller Element.</strong></p><table><thead><tr><th></th><th>Muller A</th><th>Muller D</th><th>Muller E</th></tr></thead><tbody><tr><td><b>conserved</b></td><td>1,079</td><td>1,235</td><td>1,735</td></tr><tr><td><b>recent_conserved</b></td><td>246</td><td>337</td><td>291</td></tr><tr><td><b>moved_on</b></td><td>7</td><td>4</td><td>4</td></tr><tr><td><b>gene_death</b></td><td>375</td><td>575</td><td>647</td></tr><tr><td><b>moved_off</b></td><td>135</td><td>158</td><td>193</td></tr><tr><td><b>other</b></td><td>284</td><td>386</td><td>468</td></tr></tbody></table><div class=book-tabs><input type=radio name=tabs-muller_arms style=display:none id=tabs-muller_arms-0 checked>
<label for=tabs-muller_arms-0>Muller A</label><div class="book-tabs-content markdown-inner"><figure><img src=./2019-07-22_neox_analysis_boxplot_mullerA.svg alt="Figure 2. Gene movement on/off of Muller element A. Selection Favor vs Selection Against: Mann-Whitney-U p-value: 1.32e-05" width=100%><figcaption><p><b>Figure 2. Gene movement on/off of Muller element A.</b> Selection Favor vs Selection Against: Mann-Whitney-U p-value: 1.32e-05</p></figcaption></figure><p><strong>Table 2. Number of genes with differential expression on Muller A.</strong></p><table><thead><tr><th>DEG</th><th>conserved</th><th>moved_on</th><th>gene_death</th><th>moved_off</th></tr></thead><tbody><tr><td>NS</td><td>0<sup>*</sup></td><td>1<sup>‚Ä†</sup></td><td>4</td><td>1</td></tr><tr><td>cyte</td><td>1<sup>*</sup></td><td>1</td><td>9<sup>‚Ä†</sup></td><td>1</td></tr><tr><td>gonia</td><td>62<sup>‚Ä†</sup></td><td>0<sup>*</sup></td><td>19<sup>*</sup></td><td>4</td></tr></tbody></table><p><span style=font-size:.8em>ùõò^2: 35.4513, p-value: 0.0000, df: 6,
<sup>*</sup>Significantly depleted (p-value ‚â§ 0.01),
<sup>‚Ä†</sup>Significantly enriched (p-value ‚â§ 0.01)</span></p><p><strong>Table 3. Number of genes with selection on Muller A.</strong></p><table><thead><tr><th>DEG</th><th>Selection Favor</th><th>Selection Against</th></tr></thead><tbody><tr><td>Cyte Biased</td><td>2</td><td>10</td></tr><tr><td>Not Biased</td><td>63</td><td>28</td></tr></tbody></table><ul><li>Fisher&rsquo;s exact test p-value: 0.0007</li></ul></div><input type=radio name=tabs-muller_arms style=display:none id=tabs-muller_arms-1>
<label for=tabs-muller_arms-1>Muller D</label><div class="book-tabs-content markdown-inner"><figure><img src=./2019-07-22_neox_analysis_boxplot_mullerD.svg alt="Figure 3. Gene movement on/off of Muller element D. Selection Favor vs Selection Against: Mann-Whitney-U p-value: 4.78e-12" width=100%><figcaption><p><b>Figure 3. Gene movement on/off of Muller element D.</b> Selection Favor vs Selection Against: Mann-Whitney-U p-value: 4.78e-12</p></figcaption></figure><p><strong>Table 4. Number of genes with differential expression on Muller D.</strong></p><table><thead><tr><th>DEG</th><th>measure</th><th>conserved</th><th>gene_death</th><th>moved_off</th></tr></thead><tbody><tr><td>NS</td><td>observed</td><td>8</td><td>7</td><td>1</td></tr><tr><td>cyte</td><td>observed</td><td>17<sup>*</sup></td><td>27<sup>‚Ä†</sup></td><td>7</td></tr><tr><td>gonia</td><td>observed</td><td>86<sup>‚Ä†</sup></td><td>7<sup>*</sup></td><td>4</td></tr></tbody></table><p><span style=font-size:.8em>ùõò^2: 50.6224, p-value: 0.0000, df: 4,
<sup>*</sup>Significantly depleted (p-value ‚â§ 0.01),
<sup>‚Ä†</sup>Significantly enriched (p-value ‚â§ 0.01)</span></p><p><strong>Table 5. Number of genes with selection on Muller D.</strong></p><table><thead><tr><th>DEG</th><th>Selection Favor</th><th>Selection Against</th></tr></thead><tbody><tr><td>Cyte Biased</td><td>17</td><td>34</td></tr><tr><td>Not Biased</td><td>94</td><td>19</td></tr></tbody></table><p><span style=font-size:.8em>Fisher&rsquo;s exact test p-value: 6.461e-10
Mann-Whitney-U p-value: ‚â§ 0.001</span></p></div><input type=radio name=tabs-muller_arms style=display:none id=tabs-muller_arms-2>
<label for=tabs-muller_arms-2>Muller E</label><div class="book-tabs-content markdown-inner"><figure><img src=./2019-07-22_neox_analysis_boxplot_mullerE.svg alt="Figure 4. Gene movement on/off of Muller element D. Selection Favor vs Selection Against: Mann-Whitney-U p-value: 3.75e-17" width=100%><figcaption><p><b>Figure 4. Gene movement on/off of Muller element D.</b> Selection Favor vs Selection Against: Mann-Whitney-U p-value: 3.75e-17</p></figcaption></figure><p><strong>Table 6. Number of genes with differential expression on Muller E.</strong></p><table><thead><tr><th>DEG</th><th>measure</th><th>conserved</th><th>moved_on</th><th>gene_death</th><th>moved_off</th></tr></thead><tbody><tr><td>NS</td><td>observed</td><td>13</td><td>0</td><td>5</td><td>2</td></tr><tr><td>cyte</td><td>observed</td><td>28<sup>*</sup></td><td>1</td><td>28<sup>‚Ä†</sup></td><td>10</td></tr><tr><td>gonia</td><td>observed</td><td>117<sup>‚Ä†</sup></td><td>0</td><td>12<sup>*</sup></td><td>11</td></tr></tbody></table><p><span style=font-size:.8em>ùõò^2: 41.5756, p-value: 0.0000, df: 6,
<sup>*</sup>Significantly depleted (p-value ‚â§ 0.01),
<sup>‚Ä†</sup>Significantly enriched (p-value ‚â§ 0.01)</span></p><p><strong>Table 7. Number of genes with selection on Muller E.</strong></p><table><thead><tr><th>DEG</th><th>Selection Favor</th><th>Selection Against</th></tr></thead><tbody><tr><td>Cyte Biased</td><td>29</td><td>38</td></tr><tr><td>Not Biased</td><td>130</td><td>30</td></tr></tbody></table><p><span style=font-size:.8em>Fisher&rsquo;s exact test p-value: 3.264e-08</span></p></div></div><p></p></article><div class="book-footer justify-between"><div><a href=https://github.com/jfear/larval_gonad_docs/commit/fd51e1df2faf3f5155b6b22d2f8e8fb342927d6f title="Last modified Jul 22, 2019 by Justin Fear" target=_blank rel=noopener><img src=/larval_gonad_docs/svg/calendar.svg alt=Changed> Jul 22, 2019</a></div><div><a href=https://github.com/jfear/larval_gonad_docs/edit/master/content/docs/gene_movement/_index.md target=_blank rel=noopener><img src=/larval_gonad_docs/svg/edit.svg alt=Edit> Edit this page</a></div></div></div><aside class="book-toc level-4 fixed"><nav id=TableOfContents><ul><li><ul><li><a href=#neo-x-gene-movement>Neo X Gene Movement</a><ul><li><a href=#methods>Methods</a><ul><li><a href=#identification-of-orthologs>Identification of orthologs</a></li><li><a href=#differential-expression>Differential expression</a></li></ul></li><li><a href=#results>Results</a></li></ul></li></ul></li></ul></nav></aside></main></body></html>