<!doctype html><html lang=en><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><title>Cell Clustering
| Larval Testis X Inactivation</title><link rel=stylesheet href=/larval_gonad_docs/book.min.c4ee552cc972ef71d5af01635e3098aaecd8beebdbc742ba939e26a7d7a86b0e.css><link rel=icon href=/larval_gonad_docs/favicon.png type=image/x-icon><link rel=alternate type=application/rss+xml href=https://jfear.github.io/larval_gonad_docs/docs/scrnaseq_prep/clustering/index.xml title="Larval Testis X Inactivation"></head><body><input type=checkbox style=display:none id=menu-control><main class="flex container"><aside class="book-menu fixed"><nav><h2 class=book-brand><a href=https://jfear.github.io/larval_gonad_docs>Larval Testis X Inactivation</a></h2><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/>Data Prep and Quality Control</a><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cell_selection/>Cell Selection</a><ul></ul></li><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/clustering/ class=active>Cell Clustering</a><ul></ul></li></ul></li></ul></nav><script>(function(){var menu=document.querySelector('aside.book-menu nav')
addEventListener('beforeunload',function(event){localStorage.setItem('menu.scrollTop',menu.scrollTop)});menu.scrollTop=localStorage.getItem('menu.scrollTop')})()</script></aside><div class=book-page><header class="align-center justify-between book-header"><label for=menu-control><img src=/larval_gonad_docs/svg/menu.svg alt=Menu></label>
<strong>Cell Clustering</strong></header><article class=markdown><p><span style="font-size:3em;font-weight:'strong';color:red">WORK IN PROGRESS DO NOT READ</span></p><h1 id=cell-clustering>Cell Clustering</h1><p>Cell clustering is the next step after demultiplexing and alignment.
There are a number of tools created for scRNA-Seq analysis.
The two most popular tools are <a href=https://satijalab.org/seurat/><code>Seurat</code></a> and <a href=http://cole-trapnell-lab.github.io/monocle-release/><code>Monocle</code></a>.
Both tools are very similar and follow the generate approach.</p><ol><li>Cells are further filtered based on gene number and low/upper UMI cutoffs.</li><li>Dimensionality is reduced.<ul><li>This can include gene filtering (features)</li><li>Also PCA like approaches to selected <em>N</em> orthogonal dimensions.</li></ul></li><li>A K Nearest Neighbors (KNN) graph is constructed.</li><li>The <a href=https://en.wikipedia.org/wiki/Louvain_Modularity>Louvain algorithm</a>, or something similar, is used to identify sub module in the graph to call clusters (communities)</li><li>Differential expression among clusters helps to annotate clusters as a specific cell type.</li></ol><p>None of the steps are deterministic.
We expect slight differences among approaches as well as slight different even among runs.
Below I summarize results from different methods and parameterization.</p><p>Currently there are no methods that correctly model scRNA-Seq replicates.
The best method I could find was using <a href=https://en.wikipedia.org/wiki/Canonical_correlation>canonical correlation analysis (CCA)</a>.
CCA tries to find the vectors that maximize the correlation among two data sets.
<code>Seurat</code> implements a form of CCA that can handle multiple samples.
We use CCA to &ldquo;align&rdquo; our three replicates and then call clusters on the combined cells.</p><h2 id=seurat-lower-cutoff>Seurat (lower cutoff)</h2><p><a href=https://github.com/jfear/larval_gonad/blob/5f8f4569ea253ab96d497055e7ae6ebb4c82a744/scrnaseq-wf/scripts/scrnaseq_combine_force.Rmd target=_blank class=book-btn>Code</a></p><p>For each replicate, I removed genes that were no expressed in ≥3 cells and I removed cells that did not have ≥200 expressed genes.
I then identified variable genes in each replicate using <a href=https://github.com/satijalab/seurat/blob/64e73ee19d4462aaa7a1b51cc62d113438bf2f3f/R/preprocessing.R#L677-L850><code>Seurat::FindVariableGenes</code></a>.</p><table><thead><tr><th>Replicate</th><th>Number Variable Genes</th></tr></thead><tbody><tr><td>1</td><td>576</td></tr><tr><td>2</td><td>494</td></tr><tr><td>3</td><td>330</td></tr><tr><td>1 ∩ 2 ∩ 3</td><td>108</td></tr></tbody></table><p>Next I ran CCA using the 108 common variable genes.
Using 25 CCA dimensions I aligned the three replicates.
Finally I use <a href=https://github.com/satijalab/seurat/blob/64e73ee19d4462aaa7a1b51cc62d113438bf2f3f/R/cluster_determination.R#L3-L178><code>Seurat::FindClusters</code></a> to call clusters in 25 dimension Aligned CCA space.
I used three different resolutions for clustering finding (0.4, 0.6, 1).</p><p>We decided to use <code>res=0.6</code> for downstream analysis.</p><figure><img src=tsne.svg width=100%></figure><h3 id=resolution-0-4>Resolution 0.4</h3><figure><img src=tsne_seurat_low_res0.4.png></figure><p>When using a resolution of 0.4 we obtained 9 clusters.</p><h3 id=resolution-0-6>Resolution 0.6</h3><figure><img src=tsne_seurat_low_res0.6.png></figure><h3 id=resolution-1>Resolution 1</h3><figure><img src=tsne_seurat_low_res1.png></figure><h2 id=seurat-upper-cutoff>Seurat (upper cutoff)</h2><p><a href=https://github.com/jfear/larval_gonad/blob/10050d3a5720091778329e72f3734a4fba86625e/scrnaseq-wf/scripts/scrnaseq_combine_force.Rmd#L60-L64 target=_blank class=book-btn>Code</a></p><h2 id=monocle>Monocle ()</h2><p><a class=book-btn>Code</a></p></article><div class="book-footer justify-between"><div><a href=https://github.com/jfear/larval_gonad_docs/commit/031ab5ab3c8bb8ec1bd6d78cceb1fe96c0a2a457 title="Last modified Jun 11, 2019 by Justin M. Fear" target=_blank rel=noopener><img src=/larval_gonad_docs/svg/calendar.svg alt=Changed> Jun 11, 2019</a></div><div><a href=https://github.com/jfear/larval_gonad_docs/edit/master/content/docs/scrnaseq_prep/clustering/_index.md target=_blank rel=noopener><img src=/larval_gonad_docs/svg/edit.svg alt=Edit> Edit this page</a></div></div></div><aside class="book-toc level-4 fixed"><nav id=TableOfContents><ul><li><a href=#cell-clustering>Cell Clustering</a><ul><li><a href=#seurat-lower-cutoff>Seurat (lower cutoff)</a><ul><li><a href=#resolution-0-4>Resolution 0.4</a></li><li><a href=#resolution-0-6>Resolution 0.6</a></li><li><a href=#resolution-1>Resolution 1</a></li></ul></li><li><a href=#seurat-upper-cutoff>Seurat (upper cutoff)</a></li><li><a href=#monocle>Monocle ()</a></li></ul></li></ul></nav></aside></main></body></html>