<!doctype html><html lang=en><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><title>D. willistoni (SP vs PS)
| Larval Testis X Inactivation</title><link rel=stylesheet href=/larval_gonad_docs/book.min.e0adbac1e0fc6fdc3ef34548c686f58a7e1ba9fb1adf5c9591aa4b7df3653f22.css><link rel=icon href=/larval_gonad_docs/favicon.png type=image/x-icon></head><body><input type=checkbox style=display:none id=menu-control><main class="flex container"><aside class="book-menu fixed"><nav><h2 class=book-brand><a href=https://jfear.github.io/larval_gonad_docs>Larval Testis X Inactivation</a></h2><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/>Data Prep and Quality Control</a><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cell_selection/>Cell Selection</a><ul></ul></li><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cluster_selection/>Selecting Clustering Algorithm</a><ul><li><a href=/larval_gonad_docs/docs/scrnaseq_prep/cluster_selection/seurat2/testis1/>Seurat2 Testis 1 Clustering</a></li></ul></li></ul></li><li><a href=/larval_gonad_docs/docs/clustering/>Clustering</a><ul><li><a href=/larval_gonad_docs/docs/clustering/combined_n3/>Integrated Clustering (n = 3)</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis1/>Testis 1 Individual Clustering</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis2/>Testis 2 Individual Clustering</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis3/>Testis 3 Individual Clustering</a></li><li><a href=/larval_gonad_docs/docs/clustering/testis4/>Testis 4 Individual Clustering</a></li></ul></li><li><a href=/larval_gonad_docs/docs/intronless_genes/>Intronless Gene Expression</a><ul></ul></li><li><a href=/larval_gonad_docs/docs/gene_movement/>Neo X Gene Movement</a><ul><li><a href=/larval_gonad_docs/docs/gene_movement/dpse_gonia_vs_cytes/>D. Pseudoobscura (SP vs Cytes)</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dpse_gonia_vs_eps/>D. Pseudoobscura (SP vs EPS)</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dpse_gonia_vs_ps/>D. Pseudoobscura (SP vs PS)</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dpse_eps_vs_ps/>D. Pseudoobscura (EPS vs PS)</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dwil_gonia_vs_cytes/>D. willistoni (SP vs Cytes)</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dwil_gonia_vs_eps/>D. willistoni (SP vs EPS)</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dwil_gonia_vs_ps/ class=active>D. willistoni (SP vs PS)</a></li><li><a href=/larval_gonad_docs/docs/gene_movement/dwil_eps_vs_ps/>D. willistoni (EPS vs PS)</a></li></ul></li></ul></nav><script>(function(){var menu=document.querySelector('aside.book-menu nav')
addEventListener('beforeunload',function(event){localStorage.setItem('menu.scrollTop',menu.scrollTop)});menu.scrollTop=localStorage.getItem('menu.scrollTop')})()</script></aside><div class=book-page><header class="align-center justify-between book-header"><label for=menu-control><img src=/larval_gonad_docs/svg/menu.svg alt=Menu></label>
<strong>D. willistoni (SP vs PS)</strong></header><article class=markdown><h4 id=d-williostoni-sp-vs-ps>D. williostoni (SP vs PS)</h4><p>I performed differential expression analysis by comparing cells in the SP cluster with cells in PS1, PS2, PS3.
I then took the list of genes from this analysis (SP-biased, PS-biased, Not Significant) and filtered out the orthologs.
The majority of orthologs expressed in the germline were conserved with very few genes moving onto the respective Muller element (<strong>Table 1</strong>).</p><p><strong>Table 1. Number of genes per Muller Element.</strong></p><table><thead><tr><th>index</th><th>muller_A</th><th>muller_D</th><th>muller_E</th></tr></thead><tbody><tr><td>conserved</td><td>80</td><td>151</td><td>218</td></tr><tr><td>moved_on</td><td>0</td><td>0</td><td>0</td></tr><tr><td>gene_death</td><td>30</td><td>22</td><td>34</td></tr><tr><td>moved_off</td><td>2</td><td>18</td><td>6</td></tr><tr><td>other</td><td>28</td><td>34</td><td>26</td></tr></tbody></table><p>Using these conservation classes I wanted to see if there were differences in PS expression.
We can look for enrichment in two ways: (1) we can look at the percent of PS cells expressing genes in each class,
(2) we can use contingency tables to test to look for a relationship between expression bias and movement class.
If MSCI is driving gene movement, we expect that Muller A (X) and D (Neo-X) will have increased movement off for genes important for spermatogenesis.
As a control, Muller E (2) would not show selection.</p><h3 id=ps-expression>PS Expression</h3><p>In <strong>Figures 1-3</strong> I look at the proportion of PS cells that expressed genes in the different movement classes.
The number of genes in the moved_on and moved_off classes are very low (&lt; 10) which makes interpretation difficult.
I decided to group genes into &ldquo;Selection Favor&rdquo; and &ldquo;Selection Against&rdquo; for all statistical analyses.
Here I compare the distribution of the proportion of cells using a Mann-Whitney U test.
<strong>We see an significant increase in PS expression for genes with &ldquo;Selection Against&rdquo; for all three Muller elements.</strong></p><div class=book-tabs><input type=radio name=tabs-cyte_dist style=display:none id=tabs-cyte_dist-0 checked>
<label for=tabs-cyte_dist-0>Muller A</label><div class="book-tabs-content markdown-inner"><h4 id=mann-whitney-u-favor-vs-against-1-247889494162566e-10>Mann-Whitney U (Favor vs Against): 1.247889494162566e-10</h4><figure><img src=../neox_analysis_boxplot_gonia_vs_ps_dwil_muller_A.svg alt="Figure 1. Gene movement on/off of Muller element A." width=100%><figcaption><p><b>Figure 1. Gene movement on/off of Muller element A.</b></p></figcaption></figure></div><input type=radio name=tabs-cyte_dist style=display:none id=tabs-cyte_dist-1>
<label for=tabs-cyte_dist-1>Muller D</label><div class="book-tabs-content markdown-inner"><h4 id=mann-whitney-u-favor-vs-against-4-356009043322584e-24>Mann-Whitney U (Favor vs Against): 4.356009043322584e-24</h4><figure><img src=../neox_analysis_boxplot_gonia_vs_ps_dwil_muller_D.svg alt="Figure 2. Gene movement on/off of Muller element D." width=100%><figcaption><p><b>Figure 2. Gene movement on/off of Muller element D.</b></p></figcaption></figure></div><input type=radio name=tabs-cyte_dist style=display:none id=tabs-cyte_dist-2>
<label for=tabs-cyte_dist-2>Muller E</label><div class="book-tabs-content markdown-inner"><h4 id=mann-whitney-u-favor-vs-against-3-009253790452862e-45>Mann-Whitney U (Favor vs Against): 3.009253790452862e-45</h4><figure><img src=../neox_analysis_boxplot_gonia_vs_ps_dwil_muller_E.svg alt="Figure 3. Gene movement on/off of Muller element E." width=100%><figcaption><p><b>Figure 3. Gene movement on/off of Muller element E.</b></p></figcaption></figure></div></div><p></p><h4 id=enrichment-of-ps-biased-genes>Enrichment of PS biased genes</h4><p>Next, we looked at contingency tables to test for a relationship between PS biased expression and selection against.
Again, cell counts are very low for some combinations (<strong>Table 2,4,6</strong>) so I combined counts to make statistical analysis possible (<strong>Table 3,5,7</strong>).
Again there is evidence of a enrichment of genes that underwent &ldquo;Selection Against&rdquo; for all tested Muller elements.</p><div class=book-tabs><input type=radio name=tabs-crosstabs style=display:none id=tabs-crosstabs-0 checked>
<label for=tabs-crosstabs-0>Muller A</label><div class="book-tabs-content markdown-inner"><p><strong>Table 2. Number of genes with differential expression on Muller A.</strong></p><table><thead><tr><th>bias</th><th>conserved</th><th>moved_on</th><th>gene_death</th><th>moved_off</th></tr></thead><tbody><tr><td>NS</td><td>0</td><td>0</td><td>7</td><td>0</td></tr><tr><td>PS</td><td>1</td><td>0</td><td>5</td><td>1</td></tr><tr><td>SP</td><td>79</td><td>0</td><td>18</td><td>1</td></tr></tbody></table><p><strong>Table 3. Number of genes with selection on Muller A.</strong></p><table><thead><tr><th>bias</th><th>Selection Favor</th><th>Selection Against</th></tr></thead><tbody><tr><td>PS Biased</td><td>1</td><td>6</td></tr><tr><td>Not Biased</td><td>79</td><td>26</td></tr></tbody></table><p>Fisher&rsquo;s Exact Test: 0.002094000366233636</p></div><input type=radio name=tabs-crosstabs style=display:none id=tabs-crosstabs-1>
<label for=tabs-crosstabs-1>Muller D</label><div class="book-tabs-content markdown-inner"><p><strong>Table 4. Number of genes with differential expression on Muller D.</strong></p><table><thead><tr><th>bias</th><th>conserved</th><th>moved_on</th><th>gene_death</th><th>moved_off</th></tr></thead><tbody><tr><td>NS</td><td>12</td><td>0</td><td>3</td><td>3</td></tr><tr><td>PS</td><td>31</td><td>0</td><td>13</td><td>10</td></tr><tr><td>SP</td><td>108</td><td>0</td><td>6</td><td>5</td></tr></tbody></table><p><strong>Table 5. Number of genes with selection on Muller D.</strong></p><table><thead><tr><th>bias</th><th>Selection Favor</th><th>Selection Against</th></tr></thead><tbody><tr><td>PS Biased</td><td>31</td><td>23</td></tr><tr><td>Not Biased</td><td>120</td><td>17</td></tr></tbody></table><p>Fisher&rsquo;s Exact Test: 1.6180277946244638e-05</p></div><input type=radio name=tabs-crosstabs style=display:none id=tabs-crosstabs-2>
<label for=tabs-crosstabs-2>Muller E</label><div class="book-tabs-content markdown-inner"><p><strong>Table 6. Number of genes with differential expression on Muller E.</strong></p><table><thead><tr><th>bias</th><th>conserved</th><th>moved_on</th><th>gene_death</th><th>moved_off</th></tr></thead><tbody><tr><td>NS</td><td>16</td><td>0</td><td>5</td><td>2</td></tr><tr><td>PS</td><td>50</td><td>0</td><td>21</td><td>3</td></tr><tr><td>SP</td><td>152</td><td>0</td><td>8</td><td>1</td></tr></tbody></table><p><strong>Table 7. Number of genes with selection on Muller E.</strong></p><table><thead><tr><th>bias</th><th>Selection Favor</th><th>Selection Against</th></tr></thead><tbody><tr><td>PS Biased</td><td>50</td><td>24</td></tr><tr><td>Not Biased</td><td>168</td><td>16</td></tr></tbody></table><p>Fisher&rsquo;s Exact Test: 6.496432841647948e-06</p></div></div><p></p></article><div class="book-footer justify-between"><div><a href=https://github.com/jfear/larval_gonad_docs/commit/d3b7b68a3564913f13076e4f1427404ec83523a5 title="Last modified Aug 9, 2019 by Justin M. Fear" target=_blank rel=noopener><img src=/larval_gonad_docs/svg/calendar.svg alt=Changed> Aug 9, 2019</a></div><div><a href=https://github.com/jfear/larval_gonad_docs/edit/master/content/docs/gene_movement/dwil_gonia_vs_ps.md target=_blank rel=noopener><img src=/larval_gonad_docs/svg/edit.svg alt=Edit> Edit this page</a></div></div></div><aside class="book-toc level-4 fixed"><nav id=TableOfContents><ul><li><ul><li><ul><li><ul><li><a href=#d-williostoni-sp-vs-ps>D. williostoni (SP vs PS)</a></li></ul></li><li><a href=#ps-expression>PS Expression</a><ul><li><a href=#enrichment-of-ps-biased-genes>Enrichment of PS biased genes</a></li></ul></li></ul></li></ul></li></ul></nav></aside></main></body></html>